---
title: æ–‡çŒ®-ç”¨å›¾å·ç§¯ç½‘ç»œæ¨¡æ‹Ÿå¤šè¯å‰¯ä½œç”¨
author: Package Build
date: '2021-09-29'
slug: æ–‡çŒ®-ç”¨å›¾å·ç§¯ç½‘ç»œæ¨¡æ‹Ÿå¤šè¯å‰¯ä½œç”¨
categories:
  - å­¦ä¹ 
tags:
  - æ–‡çŒ®
---

[åŸæ–‡](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770)

## Abstract

### Motivation

The use of drug combinations, termed polypharmacy, is common to treat patients with complex diseases or co-existing conditions. However, a major consequence of polypharmacy is a much higher risk of adverse side effects for the patient. Polypharmacy side effects emerge because of drugâ€“drug interactions, in which activity of one drug may change, favorably or unfavorably, if taken with another drug. The knowledge of drug interactions is often limited because these complex relationships are rare, and are usually not observed in relatively small clinical testing. Discovering polypharmacy side effects thus remains an important challenge with significant implications for patient mortality and morbidity.

è”åˆç”¨è¯ï¼Œç§°ä¸ºå¤šè¯ç–—æ³•ï¼Œé€šå¸¸ç”¨äºæ²»ç–—å¤æ‚ç–¾ç—…æˆ–å¹¶å­˜ç–¾ç—…çš„æ‚£è€…ã€‚ç„¶è€Œï¼Œå¤šè¯ç–—æ³•çš„ä¸€ä¸ªä¸»è¦åæœæ˜¯æ‚£è€…äº§ç”Ÿä¸è‰¯å‰¯ä½œç”¨çš„é£é™©æ›´é«˜ã€‚å¤šè¯å‰¯ä½œç”¨çš„äº§ç”Ÿæ˜¯å› ä¸ºè¯ç‰©ä¹‹é—´çš„ç›¸äº’ä½œç”¨ï¼Œå…¶ä¸­ä¸€ç§è¯ç‰©çš„æ´»æ€§å¯èƒ½ä¼šå‘ç”Ÿå˜åŒ–ï¼Œå¦‚æœä¸å¦ä¸€ç§è¯ç‰©ä¸€èµ·æœç”¨ï¼Œè¿™ç§å˜åŒ–å¯èƒ½æ˜¯æœ‰åˆ©çš„ï¼Œä¹Ÿå¯èƒ½æ˜¯ä¸åˆ©çš„ã€‚è¯ç‰©ç›¸äº’ä½œç”¨çš„çŸ¥è¯†é€šå¸¸æ˜¯æœ‰é™çš„ï¼Œå› ä¸ºè¿™äº›å¤æ‚çš„å…³ç³»æ˜¯ç½•è§çš„ï¼Œå¹¶ä¸”é€šå¸¸åœ¨ç›¸å¯¹è¾ƒå°çš„ä¸´åºŠè¯•éªŒä¸­æ²¡æœ‰è§‚å¯Ÿåˆ°ã€‚å› æ­¤ï¼Œå‘ç°å¤šè¯å‰¯ä½œç”¨ä»ç„¶æ˜¯ä¸€ä¸ªé‡è¦çš„æŒ‘æˆ˜ï¼Œå¯¹æ‚£è€…æ­»äº¡ç‡å’Œå‘ç—…ç‡å…·æœ‰é‡å¤§å½±å“ã€‚

### Results

Here, we present Decagon, an approach for modeling polypharmacy side effects. The approach constructs a multimodal graph of proteinâ€“protein interactions, drugâ€“protein target interactions and the polypharmacy side effects, which are represented as drugâ€“drug interactions, where each side effect is an edge of a different type. Decagon is developed specifically to handle such multimodal graphs with a large number of edge types. Our approach develops a new graph convolutional neural network for multirelational link prediction in multimodal networks. Unlike approaches limited to predicting simple drugâ€“drug interaction values, Decagon can predict the exact side effect, if any, through which a given drug combination manifests clinically. Decagon accurately predicts polypharmacy side effects, outperforming baselines by up to 69%. We find that it automatically learns representations of side effects indicative of co-occurrence of polypharmacy in patients. Furthermore, Decagon models particularly well polypharmacy side effects that have a strong molecular basis, while on predominantly non-molecular side effects, it achieves good performance because of effective sharing of model parameters across edge types. Decagon opens up opportunities to use large pharmacogenomic and patient population data to flag and prioritize polypharmacy side effects for follow-up analysis via formal pharmacological studies.

åœ¨è¿™é‡Œï¼Œæˆ‘ä»¬ä»‹ç»äº†Decagonï¼Œä¸€ç§æ¨¡æ‹Ÿå¤šè¯å‰¯ä½œç”¨çš„æ–¹æ³•ã€‚è¯¥æ–¹æ³•æ„å»ºäº†è›‹ç™½è´¨-è›‹ç™½è´¨ç›¸äº’ä½œç”¨ã€è¯ç‰©-è›‹ç™½è´¨é¶ç‚¹ç›¸äº’ä½œç”¨å’Œå¤šè¯å‰¯ä½œç”¨çš„å¤šæ¨¡æ€å›¾è°±ï¼Œè¿™äº›å‰¯ä½œç”¨è¡¨ç°ä¸ºè¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ï¼Œå…¶ä¸­æ¯ç§å‰¯ä½œç”¨éƒ½æ˜¯ä¸åŒç±»å‹çš„è¾¹ç¼˜ã€‚Decagonæ˜¯ä¸“é—¨ä¸ºå¤„ç†å…·æœ‰å¤§é‡è¾¹ç±»å‹çš„å¤šæ¨¡æ€å›¾è°±è€Œå¼€å‘çš„ã€‚æˆ‘ä»¬çš„æ–¹æ³•å¼€å‘äº†ä¸€ç§æ–°çš„å›¾å·ç§¯ç¥ç»ç½‘ç»œï¼Œç”¨äºå¤šæ¨¡ç½‘ç»œä¸­çš„å¤šå…³ç³»é“¾è·¯é¢„æµ‹ã€‚ä¸ä»…é™äºé¢„æµ‹ç®€å•è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨çš„æ–¹æ³•ä¸åŒï¼ŒDecagonå¯ä»¥é¢„æµ‹ç‰¹å®šè¯ç‰©ç»„åˆä¸´åºŠè¡¨ç°çš„ç¡®åˆ‡å‰¯ä½œç”¨ï¼ˆå¦‚æœæœ‰ï¼‰ã€‚Decagonå‡†ç¡®é¢„æµ‹äº†å¤šè¯å‰¯ä½œç”¨ï¼Œæ¯”åŸºçº¿é«˜å‡º69%ã€‚æˆ‘ä»¬å‘ç°ï¼Œå®ƒèƒ½è‡ªåŠ¨å­¦ä¹ è¡¨æ˜æ‚£è€…åŒæ—¶å‡ºç°å¤šç§è¯ç‰©å‰¯ä½œç”¨çš„è¡¨å¾ã€‚æ­¤å¤–ï¼ŒDecagonæ¨¡å‹ç‰¹åˆ«é€‚åˆå…·æœ‰å¼ºå¤§åˆ†å­åŸºç¡€çš„å¤šè¯å‰¯ä½œç”¨ï¼Œè€Œå¯¹äºä¸»è¦çš„éåˆ†å­å‰¯ä½œç”¨ï¼Œç”±äºåœ¨è¾¹ç¼˜ç±»å‹ä¹‹é—´æœ‰æ•ˆå…±äº«æ¨¡å‹å‚æ•°ï¼Œå®ƒå®ç°äº†è‰¯å¥½çš„æ€§èƒ½ã€‚Decagonæä¾›äº†åˆ©ç”¨å¤§é‡è¯ç‰©åŸºå› ç»„å­¦å’Œæ‚£è€…ç¾¤ä½“æ•°æ®æ ‡è®°å¹¶ä¼˜å…ˆè€ƒè™‘å¤šè¯å‰¯ä½œç”¨çš„æœºä¼šï¼Œä»¥ä¾¿é€šè¿‡æ­£å¼çš„è¯ç†å­¦ç ”ç©¶è¿›è¡Œåç»­åˆ†æã€‚

### å¯ç”¨æ€§å’Œå®æ–½

[æºä»£ç å’Œé¢„å¤„ç†æ•°æ®é›†](http://snap.stanford.edu/decagon)

## Introduction

Most human diseases are caused by complex biological processes that are resistant to the activity of any single drug (Jia *et al.*, 2009; Han *et al.*, 2017). A promising strategy to combat diseases is polypharmacy, a type of combinatorial therapy that involves a concurrent use of multiple medications, also termed a drug combination (Bansal *et al.*, 2014). A drug combination consists of multiple drugs, each of which has generally been used as a single effective medication in a patient population. Since drugs in a drug combination can modulate the activity of distinct proteins, drug combinations can improve therapeutic efficacy by overcoming the redundancy in underlying biological processes (Sun *et al.*, 2015). For example, a drug combination of Venetoclax and Idasanutlin has recently been shown to lead to superior antileukemic efficacy in the treatment of acute myeloid leukemia (Pan *et al.*, 2017). Here, the two drugs work in reciprocal ways: Venetoclax inhibits antiapoptotic Bcl-2 family proteins while Idasanutlin activates the p53 pathway, and therefore, the combination of these two drugs improves survival by simultaneously targeting complementary mechanisms (Pan *et al.*, 2017).

å¤§å¤šæ•°äººç±»ç–¾ç—…éƒ½æ˜¯ç”±å¤æ‚çš„ç”Ÿç‰©è¿‡ç¨‹å¼•èµ·çš„ï¼Œè¿™äº›ç”Ÿç‰©è¿‡ç¨‹å¯¹ä»»ä½•å•ä¸€è¯ç‰©çš„æ´»æ€§éƒ½å…·æœ‰è€è¯æ€§ï¼ˆJia*ç­‰äººï¼Œ2009å¹´ï¼›Han*ç­‰äººï¼Œ2017å¹´ï¼‰ã€‚å¯¹æŠ—ç–¾ç—…çš„ä¸€ä¸ªæœ‰å¸Œæœ›çš„ç­–ç•¥æ˜¯å¤šè¯ç–—æ³•ï¼Œè¿™æ˜¯ä¸€ç§ç»„åˆç–—æ³•ï¼ŒåŒ…æ‹¬åŒæ—¶ä½¿ç”¨å¤šç§è¯ç‰©ï¼Œä¹Ÿç§°ä¸ºè¯ç‰©ç»„åˆï¼ˆBansal*ç­‰äººï¼Œ2014å¹´ï¼‰ã€‚è¯ç‰©ç»„åˆç”±å¤šç§è¯ç‰©ç»„æˆï¼Œæ¯ç§è¯ç‰©é€šå¸¸åœ¨æ‚£è€…ç¾¤ä½“ä¸­ä½œä¸ºå•ä¸€æœ‰æ•ˆè¯ç‰©ä½¿ç”¨ã€‚ç”±äºè¯ç‰©ç»„åˆä¸­çš„è¯ç‰©å¯ä»¥è°ƒèŠ‚ä¸åŒè›‹ç™½è´¨çš„æ´»æ€§ï¼Œå› æ­¤è¯ç‰©ç»„åˆå¯ä»¥é€šè¿‡å…‹æœæ½œåœ¨ç”Ÿç‰©è¿‡ç¨‹ä¸­çš„å†—ä½™æ¥æé«˜æ²»ç–—æ•ˆæœï¼ˆSunç­‰äººï¼Œ2015å¹´ï¼‰ã€‚ä¾‹å¦‚ï¼ŒVenetoclaxå’ŒIdasanutlinçš„è”åˆç”¨è¯æœ€è¿‘è¢«è¯æ˜åœ¨æ²»ç–—æ€¥æ€§é«“ç³»ç™½è¡€ç—…æ–¹é¢å…·æœ‰ä¼˜è¶Šçš„æŠ—ç™½è¡€ç—…ç–—æ•ˆï¼ˆPan*ç­‰äººï¼Œ2017å¹´ï¼‰ã€‚åœ¨è¿™é‡Œï¼Œè¿™ä¸¤ç§è¯ç‰©ä»¥ç›¸äº’ä½œç”¨çš„æ–¹å¼å‘æŒ¥ä½œç”¨ï¼šVenetoclaxæŠ‘åˆ¶æŠ—å‡‹äº¡Bcl-2å®¶æ—è›‹ç™½ï¼Œè€ŒIdasanutlinæ¿€æ´»p53é€šè·¯ï¼Œå› æ­¤ï¼Œè¿™ä¸¤ç§è¯ç‰©çš„ç»„åˆé€šè¿‡åŒæ—¶é’ˆå¯¹äº’è¡¥æœºåˆ¶æé«˜ç”Ÿå­˜ç‡ï¼ˆPan*ç­‰äººï¼Œ2017å¹´ï¼‰ã€‚

While the use of multiple drugs may be a good practice for the treatment of many diseases (Liebler and Guengerich, 2005; Tatonetti *et al.*, 2012), a major consequence of polypharmacy to a patient is a much higher risk of side effects which are due to drugâ€“drug interactions. *Polypharmacy side effects* are difficult to identify manually because they are rare, it is practically impossible to test all possible pairs of drugs, and side effects are usually not observed in relatively small clinical testing (Bansal *et al.*, 2014; Tatonetti *et al.*, 2012). Furthermore, polypharmacy is recognized as an increasingly serious problem in the health care system affecting nearly 15% of the U.S. population (Kantor *et al.*, 2015), and costing >$177 billion a year in the U.S. in treating polypharmacy side effects (Ernst and Grizzle, 2001).

è™½ç„¶ä½¿ç”¨å¤šç§è¯ç‰©å¯èƒ½æ˜¯æ²»ç–—è®¸å¤šç–¾ç—…çš„è‰¯å¥½å®è·µï¼ˆLieblerå’ŒGuengerichï¼Œ2005å¹´ï¼›Tatoneti*ç­‰äººï¼Œ2012å¹´ï¼‰ï¼Œä½†å¤šç§è¯ç‰©å¯¹æ‚£è€…çš„ä¸»è¦åæœæ˜¯è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨äº§ç”Ÿå‰¯ä½œç”¨çš„é£é™©æ›´é«˜*ã€‚å¤šè¯å‰¯ä½œç”¨*å¾ˆéš¾æ‰‹åŠ¨è¯†åˆ«ï¼Œå› ä¸ºå®ƒä»¬å¾ˆç½•è§ï¼Œå‡ ä¹ä¸å¯èƒ½æµ‹è¯•æ‰€æœ‰å¯èƒ½çš„è¯ç‰©å¯¹ï¼Œå¹¶ä¸”é€šå¸¸åœ¨ç›¸å¯¹è¾ƒå°çš„ä¸´åºŠè¯•éªŒä¸­æœªè§‚å¯Ÿåˆ°å‰¯ä½œç”¨ï¼ˆBansal*ç­‰ï¼Œ2014ï¼›Tatoneti*ç­‰ï¼Œ2012ï¼‰ã€‚æ­¤å¤–ï¼Œåœ¨åŒ»ç–—ä¿å¥ç³»ç»Ÿä¸­ï¼Œå¤šè¯è¢«è®¤ä¸ºæ˜¯ä¸€ä¸ªæ—¥ç›Šä¸¥é‡çš„é—®é¢˜ï¼Œå½±å“åˆ°è¿‘15%çš„ç¾å›½äººå£ï¼ˆKantor*ç­‰äººï¼Œ2015å¹´ï¼‰ï¼Œåœ¨ç¾å›½ï¼Œæ²»ç–—å¤šè¯å‰¯ä½œç”¨æ¯å¹´èŠ±è´¹è¶…è¿‡1770äº¿ç¾å…ƒï¼ˆErnstå’ŒGrizzleï¼Œ2001å¹´ï¼‰ã€‚

*In vitro* experiments and clinical trials can be performed to identify drugâ€“drug interactions (Li *et al.*, 2016; Ryall and Tan, 2015), but systematic combinatorial screening of drugâ€“drug interaction candidates remains challenging and expensive (Bansal *et al.*, 2014). Researchers have thus attempted to collect drugâ€“drug interactions from scientific literature and electronic medical records (Percha *et al.*, 2012; Vilar *et al.*, 2017), and also discovered them through network modeling, analysis of molecular target signatures (Chen *et al.*, 2016a; Huang *et al.*, 2014b; Lewis *et al.*, 2015; Sun *et al.*, 2015; Takeda *et al.*, 2017), statistical association-based models and semi-supervised learning (Chen *et al.*, 2016b; Huang *et al.*, 2014a; Shi *et al.*, 2017; Zhao *et al.*, 2011) (see related work in Section 7). While these approaches can be useful to derive broad rules for describing drug interaction at the cellular level, they cannot directly guide strategies for drug combination treatments.

*å¯ä»¥è¿›è¡Œä½“å¤–*å®éªŒå’Œä¸´åºŠè¯•éªŒä»¥ç¡®å®šè¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ï¼ˆLi*ç­‰äººï¼Œ2016å¹´ï¼›Ryallå’ŒTanï¼Œ2015å¹´ï¼‰ï¼Œä½†è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨å€™é€‰ç‰©çš„ç³»ç»Ÿç»„åˆç­›é€‰ä»ç„¶å…·æœ‰æŒ‘æˆ˜æ€§å’Œæ˜‚è´µï¼ˆBansal*ç­‰äººï¼Œ2014å¹´ï¼‰ã€‚å› æ­¤ï¼Œç ”ç©¶äººå‘˜è¯•å›¾ä»ç§‘å­¦æ–‡çŒ®å’Œç”µå­ç—…å†ä¸­æ”¶é›†è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ï¼ˆPerca*ç­‰äººï¼Œ2012å¹´ï¼›Vilar*ç­‰äººï¼Œ2017å¹´ï¼‰ï¼Œå¹¶é€šè¿‡ç½‘ç»œå»ºæ¨¡ã€åˆ†å­é¶ç‚¹ç‰¹å¾åˆ†æå‘ç°äº†å®ƒä»¬ï¼ˆChen*ç­‰äººï¼Œ2016aï¼›Huang*ç­‰äººï¼Œ2014bï¼›Lewis*ç­‰äººï¼Œ2015ï¼›Sun*ç­‰äººï¼Œ2015ï¼›Takeda*ç­‰äººï¼Œ2017ï¼‰ï¼ŒåŸºäºç»Ÿè®¡å…³è”çš„æ¨¡å‹å’ŒåŠç›‘ç£å­¦ä¹ ï¼ˆChen*ç­‰äººï¼Œ2016bï¼›Huang*ç­‰äººï¼Œ2014aï¼›Shi*ç­‰äººï¼Œ2017ï¼›Zhao*ç­‰äººï¼Œ2011ï¼‰ï¼ˆå‚è§ç¬¬7èŠ‚ä¸­çš„ç›¸å…³å·¥ä½œï¼‰è™½ç„¶è¿™äº›æ–¹æ³•æœ‰åŠ©äºåœ¨ç»†èƒæ°´å¹³ä¸Šå¾—å‡ºæè¿°è¯ç‰©ç›¸äº’ä½œç”¨çš„å¹¿æ³›è§„åˆ™ï¼Œä½†å®ƒä»¬ä¸èƒ½ç›´æ¥æŒ‡å¯¼è¯ç‰©è”åˆæ²»ç–—çš„ç­–ç•¥ã€‚

In particular, these approaches characterize drugâ€“drug interactions through scores representing the overall probability/strength of an interaction but cannot predict the exact type of the side effect. More precisely, for drugs *i* and *j* these methods predict if their combination produces any exaggerated response *Sij* over and beyond the additive response expected under no interaction, regardless of the exact type or the number of side effects. That is, their goal is to answer a question: Sijâ‰ ?{}â , where *Sij* is the set of all polypharmacy side effects attributed specifically to a drug pair *i*, *j* but not to either drug alone. However, it is much more important and useful to answer whether a pair of drugs *i*, *j* will interact with a given side effect of type *r*, râˆˆ?Sijâ . Even though identification of precise polypharmacy side effects is critical for improved patient care, it remains a challenging task that has not yet been studied through predictive modeling.

ç‰¹åˆ«åœ°ï¼Œè¿™äº›æ–¹æ³•é€šè¿‡è¡¨ç¤ºç›¸äº’ä½œç”¨çš„æ€»ä½“æ¦‚ç‡/å¼ºåº¦çš„åˆ†æ•°æ¥æè¿°è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ï¼Œä½†æ— æ³•é¢„æµ‹å‰¯ä½œç”¨çš„ç¡®åˆ‡ç±»å‹ã€‚æ›´å‡†ç¡®åœ°è¯´ï¼Œå¯¹äºè¯ç‰©iå’Œjï¼Œè¿™äº›æ–¹æ³•é¢„æµ‹å®ƒä»¬çš„ç»„åˆæ˜¯å¦ä¼šåœ¨æ²¡æœ‰ç›¸äº’ä½œç”¨çš„æƒ…å†µä¸‹è¶…å‡ºé¢„æœŸä¹‹å¤–çš„åŠ æ€§ååº”(additive response)ï¼Œå¹¶äº§ç”Ÿè¶…å‡ºé¢„æœŸçš„å¤¸å¼ ååº”(exaggerated response) Sijã€‚ä¹Ÿå°±æ˜¯è¯´ï¼Œä»–ä»¬çš„ç›®æ ‡æ˜¯å›ç­”ä¸€ä¸ªé—®é¢˜ï¼šSijâ‰ ?{}â , å…¶ä¸­ï¼Œ*Sij*æ˜¯æ‰€æœ‰å¤šè¯å‰¯ä½œç”¨çš„é›†åˆï¼Œå…·ä½“å½’å› äºè¯ç‰©å¯¹*i*ï¼Œ*j*ï¼Œä½†ä¸æ˜¯å•ç‹¬å½’å› äºä»»ä½•ä¸€ç§è¯ç‰©ã€‚ç„¶è€Œï¼Œå›ç­”ä¸€å¯¹è¯ç‰© iã€j æ˜¯å¦ä¼šä¸ç»™å®šçš„ r å‹å‰¯ä½œç”¨ç›¸äº’ä½œç”¨æ›´ä¸ºé‡è¦å’Œæœ‰ç”¨ï¼Œrâˆˆ?Sijâ ã€‚å°½ç®¡ç¡®å®šç²¾ç¡®çš„å¤šè¯å‰¯ä½œç”¨å¯¹äºæ”¹å–„æ‚£è€…æŠ¤ç†è‡³å…³é‡è¦ï¼Œä½†å®ƒä»ç„¶æ˜¯ä¸€é¡¹å°šæœªé€šè¿‡é¢„æµ‹å»ºæ¨¡è¿›è¡Œç ”ç©¶çš„å…·æœ‰æŒ‘æˆ˜æ€§çš„ä»»åŠ¡ã€‚

### 1.1 Present study

Here, we develop *Decagon*, a method for predicting side effects of drug pairs. We model the problem by constructing a large two-layer multimodal graph of proteinâ€“protein interactions, drugâ€“protein interactions and drugâ€“drug interactions (i.e. side effects; Fig. 1). Each drugâ€“drug interaction is labeled by a different edge type, which signifies the type of the side effect. We then develop a new multirelational edge prediction model that uses the multimodal graph to predict drugâ€“drug interactions as well as their types. Our model is a convolutional graph neural network that operates in a multirelational setting.

åœ¨è¿™é‡Œï¼Œæˆ‘ä»¬å¼€å‘äº†â€œDecagonâ€ï¼Œä¸€ç§é¢„æµ‹è¯ç‰©å‰¯ä½œç”¨çš„æ–¹æ³•ã€‚æˆ‘ä»¬é€šè¿‡æ„å»ºè›‹ç™½è´¨-è›‹ç™½è´¨ç›¸äº’ä½œç”¨ã€è¯ç‰©-è›‹ç™½è´¨ç›¸äº’ä½œç”¨å’Œè¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ï¼ˆå³å‰¯ä½œç”¨ï¼›å›¾1ï¼‰çš„å¤§å‹ä¸¤å±‚å¤šæ¨¡æ€å›¾ï¼Œå¯¹é—®é¢˜è¿›è¡Œå»ºæ¨¡ã€‚æ¯ç§è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨éƒ½ç”¨ä¸åŒçš„è¾¹ç¼˜ç±»å‹æ ‡è®°ï¼Œè¿™è¡¨ç¤ºå‰¯ä½œç”¨çš„ç±»å‹ã€‚ç„¶åï¼Œæˆ‘ä»¬å¼€å‘äº†ä¸€ä¸ªæ–°çš„å¤šå…³ç³»è¾¹ç¼˜é¢„æµ‹æ¨¡å‹ï¼Œè¯¥æ¨¡å‹ä½¿ç”¨å¤šæ¨¡æ€å›¾è°±é¢„æµ‹è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨åŠå…¶ç±»å‹ã€‚æˆ‘ä»¬çš„æ¨¡å‹æ˜¯ä¸€ä¸ªåœ¨å¤šå…³ç³»ç¯å¢ƒä¸‹è¿è¡Œçš„å·ç§¯å›¾ç¥ç»ç½‘ç»œã€‚

![](index.assets/bty294f1.jpeg)

An example graph of polypharmacy side effects derived from genomic and patient population data. A multimodal graph consists of proteinâ€“protein interactions, drugâ€“protein targets and drugâ€“drug interactions encoded by 964 different polypharmacy side effects (i.e. edge types ri, i=1,â€¦,964). Side information is integrated into the model in the form of additional protein and drug feature vectors. Highlighted network neighbors of Ciprofloxacin (node C) indicate this drug targets four proteins and interacts with three other drugs. The graph encodes information that Ciprofloxacin (node C) taken together with Doxycycline (node D) or with Simvastatin (node S) increases the risk of bradycardia side effect (side effect type r2), and its combination with Mupirocin (M) increases the risk of gastrointestinal bleed side effect r1. We use the graph representation to develop Decagon, a graph convolutional neural model of polypharmacy side effects. Decagon predicts associations between pairs of drugs and side effects (shown in red) with the goal of identifying side effects, which cannot be attributed to either individual drug in the pair

æ¥è‡ªåŸºå› ç»„å’Œæ‚£è€…ç¾¤ä½“æ•°æ®çš„å¤šè¯å‰¯ä½œç”¨ç¤ºä¾‹å›¾ã€‚å¤šæ¨¡æ€å›¾è°±ç”±964ç§ä¸åŒçš„å¤šè¯å‰¯ä½œç”¨ç¼–ç çš„è›‹ç™½è´¨-è›‹ç™½è´¨ç›¸äº’ä½œç”¨ã€è¯ç‰©-è›‹ç™½è´¨é¶ç‚¹å’Œè¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ç»„æˆï¼ˆå³è¾¹ç¼˜ç±»å‹riï¼Œi=1ï¼Œâ€¦ï¼Œ964ï¼‰ã€‚å‰¯ä¿¡æ¯ä»¥é¢å¤–çš„è›‹ç™½è´¨å’Œè¯ç‰©ç‰¹å¾å‘é‡çš„å½¢å¼æ•´åˆåˆ°æ¨¡å‹ä¸­ã€‚çªå‡ºæ˜¾ç¤ºçš„ç¯ä¸™æ²™æ˜Ÿç½‘ç»œé‚»å±…ï¼ˆèŠ‚ç‚¹Cï¼‰è¡¨æ˜è¯¥è¯ç‰©ä»¥å››ç§è›‹ç™½è´¨ä¸ºé¶ç‚¹ï¼Œå¹¶ä¸å…¶ä»–ä¸‰ç§è¯ç‰©ç›¸äº’ä½œç”¨ã€‚å›¾ä¸­æ˜¾ç¤ºçš„ä¿¡æ¯è¡¨æ˜ï¼Œç¯ä¸™æ²™æ˜Ÿï¼ˆèŠ‚ç‚¹Cï¼‰ä¸å¼ºåŠ›éœ‰ç´ ï¼ˆèŠ‚ç‚¹Dï¼‰æˆ–è¾›ä¼ä»–æ±€ï¼ˆèŠ‚ç‚¹Sï¼‰åˆç”¨ä¼šå¢åŠ å¿ƒåŠ¨è¿‡ç¼“å‰¯ä½œç”¨ï¼ˆr2å‹å‰¯ä½œç”¨ï¼‰çš„é£é™©ï¼Œä¸è«åŒ¹ç½—æ˜Ÿï¼ˆMï¼‰åˆç”¨ä¼šå¢åŠ èƒƒè‚ é“å‡ºè¡€å‰¯ä½œç”¨çš„é£é™©ã€‚æˆ‘ä»¬ä½¿ç”¨å›¾å½¢è¡¨ç¤ºæ³•å¼€å‘äº†Decagonï¼Œä¸€ç§å¤šè¯å‰¯ä½œç”¨çš„å›¾å½¢å·ç§¯ç¥ç»æ¨¡å‹ã€‚Decagoné¢„æµ‹è¯ç‰©å¯¹å’Œå‰¯ä½œç”¨ï¼ˆçº¢è‰²æ˜¾ç¤ºï¼‰ä¹‹é—´çš„å…³è”ï¼Œç›®çš„æ˜¯è¯†åˆ«å‰¯ä½œç”¨ï¼Œè€Œå‰¯ä½œç”¨ä¸èƒ½å½’å› äºè¯ç‰©å¯¹ä¸­çš„ä»»ä½•ä¸€ç§è¯ç‰©

To motivate our model, we first perform exploratory analysis leading to two important observations (Section 3). First, we find that co-prescribed drugs (i.e. drug combinations) tend to have more target proteins in common than random drug pairs, suggesting that drug-target protein information contains valuable information for drug combination modeling. Second, we find that it is important to consider a map of proteinâ€“protein interactions in order to be able to model characteristics of drugs with common side effects. These observations motivate the development of *Decagon* to make predictions about which drug pairs will interact and what will the exact type of the interaction/side effect be (Section 4).

ä¸ºäº†æ¿€åŠ±æˆ‘ä»¬çš„æ¨¡å‹ï¼Œæˆ‘ä»¬é¦–å…ˆè¿›è¡Œæ¢ç´¢æ€§åˆ†æï¼Œå¾—å‡ºä¸¤ä¸ªé‡è¦çš„è§‚å¯Ÿç»“æœï¼ˆç¬¬3èŠ‚ï¼‰ã€‚é¦–å…ˆï¼Œæˆ‘ä»¬å‘ç°å…±åŒå¤„æ–¹è¯ç‰©ï¼ˆå³è¯ç‰©ç»„åˆï¼‰å¾€å¾€æ¯”éšæœºè¯ç‰©å¯¹å…·æœ‰æ›´å¤šå…±åŒçš„é¶è›‹ç™½ï¼Œè¿™è¡¨æ˜è¯ç‰©é¶è›‹ç™½ä¿¡æ¯åŒ…å«ç”¨äºè¯ç‰©ç»„åˆå»ºæ¨¡çš„æœ‰ä»·å€¼çš„ä¿¡æ¯ã€‚ç¬¬äºŒï¼Œæˆ‘ä»¬å‘ç°é‡è¦çš„æ˜¯è€ƒè™‘è›‹ç™½è´¨-è›‹ç™½è´¨ç›¸äº’ä½œç”¨çš„å›¾è°±ï¼Œä»¥ä¾¿èƒ½å¤Ÿæ¨¡æ‹Ÿå…·æœ‰å…±åŒå‰¯ä½œç”¨çš„è¯ç‰©çš„ç‰¹æ€§ã€‚è¿™äº›è§‚å¯Ÿç»“æœæ¨åŠ¨äº†â€œDecagonâ€çš„å‘å±•ï¼Œä»¥é¢„æµ‹å“ªäº›è¯ç‰©å¯¹å°†ç›¸äº’ä½œç”¨ï¼Œä»¥åŠç›¸äº’ä½œç”¨/å‰¯ä½œç”¨çš„ç¡®åˆ‡ç±»å‹ï¼ˆç¬¬4èŠ‚ï¼‰ã€‚

*Decagon* develops a new graph auto-encoder approach (Hamilton *et al.*, 2017a), which allows us to develop an end-to-end trainable model for link prediction on a multimodal graph. In contrast, previous graph-based approaches for link prediction tasks in biology (e.g. Chen *et al.* 2016b; Huang *et al.* 2014b; Zong *et al.*2017) employ a two-stage pipeline, typically consisting of a graph feature extraction model and a link prediction model, both of which are trained separately. Furthermore, the crucial distinguishing characteristic of *Decagon* is the *multirelational link prediction* ability allowing us to capture the interdependence of different edge (side effect) types, and to identify which out of all possible edge types exist between any two drug nodes in the graph. This is in sharp contrast with approaches for simple link prediction (Trouillon *et al.*, 2016), which predict only existence of edges between node pairs, and is also critical for modeling a large number of different edge/side effect types.

*Decagon*å¼€å‘äº†ä¸€ç§æ–°çš„**å›¾å½¢è‡ªåŠ¨ç¼–ç å™¨æ–¹æ³•**ï¼ˆHamilton*ç­‰äººã€‚*ï¼Œ2017aï¼‰ï¼Œè¯¥æ–¹æ³•å…è®¸æˆ‘ä»¬ä¸ºå¤šæ¨¡æ€å›¾å½¢ä¸Šçš„é“¾è·¯é¢„æµ‹å¼€å‘ç«¯åˆ°ç«¯å¯è®­ç»ƒæ¨¡å‹ã€‚ç›¸æ¯”ä¹‹ä¸‹ï¼Œä»¥å‰ç”¨äºç”Ÿç‰©å­¦ä¸­é“¾æ¥é¢„æµ‹ä»»åŠ¡çš„åŸºäºå›¾å½¢çš„æ–¹æ³•ï¼ˆä¾‹å¦‚ï¼ŒChen*ç­‰äººã€‚*2016bï¼›Huang*ç­‰äººã€‚*2014bï¼›Zong*ç­‰äººã€‚*2017ï¼‰é‡‡ç”¨äº†ä¸¤é˜¶æ®µç®¡é“ï¼Œé€šå¸¸ç”±å›¾ç‰¹å¾æå–æ¨¡å‹å’Œé“¾æ¥é¢„æµ‹æ¨¡å‹ç»„æˆï¼Œä¸¤è€…éƒ½æ˜¯å•ç‹¬è®­ç»ƒçš„ã€‚æ­¤å¤–ï¼Œ*Decagon*çš„å…³é”®åŒºåˆ«ç‰¹å¾æ˜¯*å¤šå…³ç³»é“¾æ¥é¢„æµ‹*èƒ½åŠ›ï¼Œä½¿æˆ‘ä»¬èƒ½å¤Ÿæ•è·ä¸åŒè¾¹ç¼˜ï¼ˆå‰¯ä½œç”¨ï¼‰ç±»å‹çš„ç›¸äº’ä¾èµ–æ€§ï¼Œå¹¶è¯†åˆ«å›¾ä¸­ä»»æ„ä¸¤ä¸ªè¯ç‰©èŠ‚ç‚¹ä¹‹é—´å­˜åœ¨çš„æ‰€æœ‰å¯èƒ½è¾¹ç¼˜ç±»å‹ä¸­çš„å“ªä¸€ç§ã€‚è¿™ä¸ç®€å•é“¾è·¯é¢„æµ‹æ–¹æ³•ï¼ˆTrouillon*ç­‰äººï¼Œ2016å¹´ï¼‰å½¢æˆé²œæ˜å¯¹æ¯”ï¼Œåè€…ä»…é¢„æµ‹èŠ‚ç‚¹å¯¹ä¹‹é—´çš„è¾¹çš„å­˜åœ¨ï¼Œå¯¹äºå»ºæ¨¡å¤§é‡ä¸åŒçš„è¾¹/å‰¯ä½œç”¨ç±»å‹ä¹Ÿè‡³å…³é‡è¦ã€‚

We contrast *Decagon*â€™s performance with that of state-of-the-art approaches for multirelational tensor factorization (Nickel *et al.*, 2011; Papalexakis *et al.*, 2017), approaches for representation learning on graphs (Perozzi *et al.*, 2014; Zong *et al.*, 2017) and established machine learning methods for link prediction, which we adapted for the polypharmacy side effect prediction task. *Decagon*outperforms alternative approaches by up to 69% and leads to a 20% average gain in predictive performance, with larger gains achieved on side effect types that have a strong molecular basis (Section 6). For several novel predictions we find supporting evidence in the biomedical literature, suggesting that *Decagon*performs especially well at identifying predictions that are highly likely to be true positive. Taken together, this study shows, for the first time, the ability to model side effects of drug combinations and opens up new opportunities for development of combinatorial drug therapies.

æˆ‘ä»¬å°†*Decagon*çš„æ€§èƒ½ä¸**å¤šå…³ç³»å¼ é‡å› å¼åˆ†è§£**çš„æœ€æ–°æ–¹æ³•ï¼ˆNickel*ç­‰ï¼Œ2011ï¼›Papalexakis*ç­‰ï¼Œ2017ï¼‰ã€**å›¾å½¢è¡¨ç¤ºå­¦ä¹ æ–¹æ³•**ï¼ˆPerozzi*ç­‰ï¼Œ2014ï¼›Zong*ç­‰ï¼Œ2017ï¼‰å’Œå»ºç«‹çš„**é“¾æ¥é¢„æµ‹æœºå™¨å­¦ä¹ **æ–¹æ³•è¿›è¡Œäº†å¯¹æ¯”ï¼Œæˆ‘ä»¬å°†å…¶ç”¨äºå¤šè¯å‰¯ä½œç”¨é¢„æµ‹ä»»åŠ¡*Decagon*çš„è¡¨ç°ä¼˜äºå…¶ä»–æ–¹æ³•é«˜è¾¾69%ï¼Œé¢„æµ‹æ€§èƒ½å¹³å‡æé«˜20%ï¼Œåœ¨å…·æœ‰å¼ºå¤§åˆ†å­åŸºç¡€çš„å‰¯ä½œç”¨ç±»å‹ä¸Šè·å¾—æ›´å¤§çš„æé«˜ï¼ˆç¬¬6èŠ‚ï¼‰ã€‚å¯¹äºä¸€äº›æ–°çš„é¢„æµ‹ï¼Œæˆ‘ä»¬åœ¨ç”Ÿç‰©åŒ»å­¦æ–‡çŒ®ä¸­æ‰¾åˆ°äº†æ”¯æŒè¯æ®ï¼Œè¡¨æ˜â€œDecagonâ€åœ¨è¯†åˆ«ææœ‰å¯èƒ½æ˜¯çœŸé˜³æ€§çš„é¢„æµ‹æ–¹é¢è¡¨ç°å¾—ç‰¹åˆ«å¥½ã€‚ç»¼ä¸Šæ‰€è¿°ï¼Œè¿™é¡¹ç ”ç©¶é¦–æ¬¡æ˜¾ç¤ºäº†æ¨¡æ‹Ÿè¯ç‰©ç»„åˆå‰¯ä½œç”¨çš„èƒ½åŠ›ï¼Œå¹¶ä¸ºç»„åˆè¯ç‰©ç–—æ³•çš„å‘å±•å¼€è¾Ÿäº†æ–°çš„æœºä¼šã€‚

## Datasets

We formulate the polypharmacy side effect identification problem as a multirelational link prediction problem in a two-layer multimodal graph/network of two node types: drugs and proteins. We construct two-layer multimodal network as follows (Fig. 1). Proteinâ€“protein interaction network describes relationships between proteins. Drugâ€“drug interaction network contains 964 different types of edges (one for each side effect type) and describes which drug pairs lead to which side effects. Lastly, drug-protein links describe the proteins targeted by a given drug.

æˆ‘ä»¬å°†å¤šè¯å‰¯ä½œç”¨è¯†åˆ«é—®é¢˜æè¿°ä¸ºè¯ç‰©å’Œè›‹ç™½è´¨ä¸¤ç§èŠ‚ç‚¹ç±»å‹çš„ä¸¤å±‚å¤šæ¨¡æ€å›¾/ç½‘ç»œä¸­çš„å¤šå…³ç³»é“¾æ¥é¢„æµ‹é—®é¢˜ã€‚æˆ‘ä»¬æ„å»ºäº†ä¸¤å±‚å¤šæ¨¡å¼ç½‘ç»œï¼Œå¦‚ä¸‹æ‰€ç¤ºï¼ˆå›¾1ï¼‰ã€‚è›‹ç™½è´¨-è›‹ç™½è´¨ç›¸äº’ä½œç”¨ç½‘ç»œæè¿°è›‹ç™½è´¨ä¹‹é—´çš„å…³ç³»ã€‚è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ç½‘ç»œåŒ…å«964ç§ä¸åŒç±»å‹çš„è¾¹ç¼˜ï¼ˆæ¯ç§å‰¯ä½œç”¨ç±»å‹ä¸€ç§ï¼‰ï¼Œå¹¶æè¿°äº†å“ªäº›è¯ç‰©å¯¹å¯¼è‡´å“ªäº›å‰¯ä½œç”¨ã€‚æœ€åï¼Œè¯ç‰©-è›‹ç™½è´¨é“¾æ¥æè¿°ç‰¹å®šè¯ç‰©çš„é¶å‘è›‹ç™½è´¨ã€‚

We continue by describing the datasets used to construct the network. Preprocessed versions of all datasets are available through this studyâ€™s website: http://snap.stanford.edu/decagon.

æˆ‘ä»¬ç»§ç»­æè¿°ç”¨äºæ„å»ºç½‘ç»œçš„æ•°æ®é›†ã€‚æ‰€æœ‰æ•°æ®é›†çš„é¢„å¤„ç†ç‰ˆæœ¬å¯é€šè¿‡æœ¬ç ”ç©¶ç½‘ç«™è·å¾—ï¼šhttp://snap.stanford.edu/decagon.

### 2.1 Proteinâ€“protein and drugâ€“protein interactions

We used the human proteinâ€“protein interaction (PPI) network compiled by Menche *et al.* (2015) and Chatr-Aryamontri *et al.* (2015), integrated with additional PPI information from Szklarczyk *et al.* (2017), and Rolland *et al.*(2014). The network contains physical interactions experimentally documented in humans, such as metabolic enzyme-coupled interactions and signaling interactions. The network is unweighted and undirected with 19â€‰085 proteins and 719â€‰402 physical interactions.

æˆ‘ä»¬ä½¿ç”¨äº†Menche*ç­‰äººï¼ˆ2015ï¼‰å’ŒChatr Aryamontri*ç­‰äººï¼ˆ2015ï¼‰ç¼–åˆ¶çš„äººç±»è›‹ç™½è´¨-è›‹ç™½è´¨ç›¸äº’ä½œç”¨ï¼ˆPPIï¼‰ç½‘ç»œï¼Œå¹¶ç»“åˆSzklarczyk*ç­‰äººï¼ˆ2017ï¼‰å’ŒRolland*ç­‰äººï¼ˆ2014ï¼‰æä¾›çš„å…¶ä»–PPIä¿¡æ¯ã€‚è¯¥ç½‘ç»œåŒ…å«åœ¨äººä½“å†…å®éªŒè®°å½•çš„ç‰©ç†ç›¸äº’ä½œç”¨ï¼Œå¦‚ä»£è°¢é…¶è€¦åˆç›¸äº’ä½œç”¨å’Œä¿¡å·ç›¸äº’ä½œç”¨ã€‚ç½‘ç»œæœªåŠ æƒä¸”æ— æ–¹å‘æ€§ï¼Œæœ‰19ä¸ªâ€‰085è›‹ç™½è´¨å’Œ719â€‰402ç‰©ç†ç›¸äº’ä½œç”¨ã€‚

We obtained relationships between proteins and drugs from the STITCH (Search Tool for InTeractions of CHemicals) database, which integrates various chemical and protein networks (Szklarczyk *et al.*, 2016). For this study, we considered only the interactions between small chemicals (i.e. drugs) and target proteins that had been experimentally verified. There were over 8â€‰083â€‰600 interactions present between 8934 proteins and 519â€‰022 chemicals.

æˆ‘ä»¬ä»STITCHï¼ˆåŒ–å­¦å“ç›¸äº’ä½œç”¨æœç´¢å·¥å…·ï¼‰æ•°æ®åº“ä¸­è·å¾—äº†è›‹ç™½è´¨å’Œè¯ç‰©ä¹‹é—´çš„å…³ç³»ï¼Œè¯¥æ•°æ®åº“é›†æˆäº†å„ç§åŒ–å­¦å’Œè›‹ç™½è´¨ç½‘ç»œï¼ˆSzklarczyk*ç­‰äººï¼Œ2016å¹´ï¼‰ã€‚åœ¨è¿™é¡¹ç ”ç©¶ä¸­ï¼Œæˆ‘ä»¬åªè€ƒè™‘äº†å°åŒ–å­¦ç‰©è´¨ï¼ˆå³è¯ç‰©ï¼‰å’Œå·²é€šè¿‡å®éªŒéªŒè¯çš„é¶è›‹ç™½ä¹‹é—´çš„ç›¸äº’ä½œç”¨ã€‚8934 ç§è›‹ç™½è´¨å’Œ 519 022 ç§åŒ–å­¦ç‰©è´¨ä¹‹é—´å­˜åœ¨è¶…è¿‡ 8 083 600 ç§ç›¸äº’ä½œç”¨ã€‚

### 2.2 Drugâ€“drug interaction and side effect data

**è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨å’Œå‰¯ä½œç”¨æ•°æ®**

We also pulled from databases detailing side effects of both individual drugs and drug combinations. The SIDER (Side Effect Resource) database contains 286â€‰399 drug-side effect associations over 1556 drugs and 5868 side effects (Kuhn *et al.*, 2016) obtained by mining adverse events from drug label text. We integrated it with the OFFSIDES database, which details off-label 487â€‰530 associations between 1332 drugs and 10â€‰097 side effects (Tatonetti *et al.*, 2012). The OFFSIDES database was generated using adverse event reporting systems that collect reports from doctors, patients and drug companies. We eliminated side effect synonyms and used one side effect vocabulary to construct all datasets. That preprocessing is important as the prediction problem would be much easier if some side effects were perfectly correlated. After combining these datasets, there is a median of 159 side effects per drug, with the most common side effects being nausea, vomiting, headache, diarrhoea and dermatitis.

æˆ‘ä»¬è¿˜ä»æ•°æ®åº“ä¸­æå–äº†è¯¦ç»†çš„ä¸ªåˆ«è¯ç‰©å’Œè¯ç‰©ç»„åˆçš„å‰¯ä½œç”¨ã€‚SIDERï¼ˆå‰¯ä½œç”¨èµ„æºï¼‰æ•°æ®åº“åŒ…å«286ä¸ªâ€‰é€šè¿‡ä»è¯ç‰©æ ‡ç­¾æ–‡æœ¬ä¸­æŒ–æ˜ä¸è‰¯äº‹ä»¶ï¼Œè·å¾—è¶…è¿‡1556ç§è¯ç‰©å’Œ5868ç§å‰¯ä½œç”¨çš„399ç§è¯ç‰©å‰¯ä½œç”¨å…³è”ï¼ˆKuhn*ç­‰äººï¼Œ2016ï¼‰ã€‚æˆ‘ä»¬å°†å…¶ä¸OFFSIDESæ•°æ®åº“é›†æˆï¼Œåè€…è¯¦ç»†ä»‹ç»äº†éæ ‡ç­¾487â€‰530 1332ç§è¯ç‰©ä¸10ç§è¯ç‰©ä¹‹é—´çš„å…³è”â€‰097å‰¯ä½œç”¨ï¼ˆTatoneti*ç­‰äººï¼Œ2012å¹´ï¼‰ã€‚è¶Šä½æ•°æ®åº“æ˜¯ä½¿ç”¨æ”¶é›†åŒ»ç”Ÿã€æ‚£è€…å’Œåˆ¶è¯å…¬å¸æŠ¥å‘Šçš„ä¸è‰¯äº‹ä»¶æŠ¥å‘Šç³»ç»Ÿç”Ÿæˆçš„ã€‚æˆ‘ä»¬æ¶ˆé™¤äº†å‰¯ä½œç”¨åŒä¹‰è¯ï¼Œå¹¶ä½¿ç”¨å‰¯ä½œç”¨è¯æ±‡è¡¨æ¥æ„å»ºæ‰€æœ‰æ•°æ®é›†ã€‚è¿™ä¸€é¢„å¤„ç†å¾ˆé‡è¦ï¼Œå› ä¸ºå¦‚æœæŸäº›å‰¯ä½œç”¨å®Œå…¨ç›¸å…³ï¼Œé¢„æµ‹é—®é¢˜å°±ä¼šå®¹æ˜“å¾—å¤šã€‚ç»¼åˆè¿™äº›æ•°æ®é›†åï¼Œæ¯ç§è¯ç‰©çš„å¹³å‡å‰¯ä½œç”¨ä¸º159ä¸ªï¼Œæœ€å¸¸è§çš„å‰¯ä½œç”¨æ˜¯æ¶å¿ƒã€å‘•åã€å¤´ç—›ã€è…¹æ³»å’Œçš®ç‚ã€‚

We pulled polypharmacy side effect information from TWOSIDES, which details 1318 side effects types across 63â€‰473 drug combinations, which are greater than expected given the effects of either drug in the combination individually (Tatonetti *et al.*, 2012). Like OFFSIDES, TWOSIDES was generated from adverse event reporting systems. Common side effects, like hypotension and nausea, occur in over a third of drug combinations, while others like amnesia and muscle spasms only occur in a handful of drug combinations. Overall, it contains 4â€‰651â€‰131 drug combination-side effect associations. In this study, we focus on predicting the 964 commonly occurring types of polypharmacy side effects that each occurred in at least 500 drug combinations.

æˆ‘ä»¬ä» TWOSIDES ä¸­æå–äº†å¤šç§è¯ç‰©å‰¯ä½œç”¨ä¿¡æ¯ï¼Œå…¶ä¸­è¯¦ç»†ä»‹ç»äº† 63 473 ç§è¯ç‰©ç»„åˆä¸­çš„ 1318 ç§å‰¯ä½œç”¨ç±»å‹ï¼Œè€ƒè™‘åˆ°ç»„åˆä¸­ä»»ä¸€è¯ç‰©çš„å•ç‹¬ä½œç”¨ï¼Œè¿™äº›å‰¯ä½œç”¨æ¯”é¢„æœŸçš„è¦å¤§ï¼ˆTatonetti ç­‰äººï¼Œ2012 å¹´ï¼‰ã€‚ä¸ OFFSIDES ä¸€æ ·ï¼ŒTWOSIDES ä¹Ÿæ˜¯ç”±ä¸è‰¯äº‹ä»¶æŠ¥å‘Šç³»ç»Ÿç”Ÿæˆçš„ã€‚å¸¸è§çš„å‰¯ä½œç”¨ï¼Œå¦‚ä½è¡€å‹å’Œæ¶å¿ƒï¼Œå‘ç”Ÿåœ¨è¶…è¿‡ä¸‰åˆ†ä¹‹ä¸€çš„è¯ç‰©ç»„åˆä¸­ï¼Œè€Œå¥å¿˜ç—‡å’Œè‚Œè‚‰ç—‰æŒ›ç­‰å…¶ä»–å‰¯ä½œç”¨åªå‘ç”Ÿåœ¨å°‘æ•°è¯ç‰©ç»„åˆä¸­ã€‚æ€»ä½“è€Œè¨€ï¼Œå®ƒåŒ…å« 4 651 131 ä¸ªè¯ç‰©ç»„åˆå‰¯ä½œç”¨å…³è”ã€‚åœ¨è¿™é¡¹ç ”ç©¶ä¸­ï¼Œæˆ‘ä»¬ä¸“æ³¨äºé¢„æµ‹ 964 ç§å¸¸è§çš„å¤šè¯å‰¯ä½œç”¨ç±»å‹ï¼Œæ¯ç§ç±»å‹è‡³å°‘åœ¨ 500 ç§è¯ç‰©ç»„åˆä¸­å‘ç”Ÿã€‚

The final network after linking entity vocabularies used by different databases has 645 drug and 19â€‰085 protein nodes connected by 715â€‰612 proteinâ€“protein, 4â€‰651â€‰131 drugâ€“drug and 18â€‰596 drugâ€“protein edges.

è¿æ¥ä¸åŒæ•°æ®åº“ä½¿ç”¨çš„å®ä½“è¯æ±‡åçš„æœ€ç»ˆç½‘ç»œæœ‰ 645 ä¸ªè¯ç‰©å’Œ 19 085 ä¸ªè›‹ç™½è´¨èŠ‚ç‚¹ï¼Œç”± 715 612 ä¸ªè›‹ç™½è´¨-è›‹ç™½è´¨ã€4 651 131 ä¸ªè¯ç‰©-è¯ç‰©å’Œ 18 596 ä¸ªè¯ç‰©-è›‹ç™½è´¨è¾¹è¿æ¥ã€‚

## 3 Data-driven motivation for *Decagon* approach

**æ•°æ®é©±åŠ¨çš„Decagonæ–¹æ³•çš„åŠ¨æœº**

Here, we make three observations about the structure of the two-layer multimodal graph (Fig. 1) that have important implications for the design of the *Decagon* model.

åœ¨è¿™é‡Œï¼Œæˆ‘ä»¬å¯¹ä¸¤å±‚å¤šæ¨¡å›¾ï¼ˆå›¾1ï¼‰çš„ç»“æ„è¿›è¡Œäº†ä¸‰æ¬¡è§‚å¯Ÿï¼Œè¿™å¯¹â€œDecagonâ€æ¨¡å‹çš„è®¾è®¡å…·æœ‰é‡è¦æ„ä¹‰ã€‚

First, we observe that there is a wide range in how frequently certain side effects occur in drug combinations. We find that >53% of polypharmacy side effects are known to occur in <3% of the documented drug combinations (e.g. cerebral artery embolism, lung abscess, sarcoma, collagen disorder). In contrast, the more frequent side effects, (e.g. vomiting, weight gain, nausea and anaemia), occur an order of magnitude more often. Due to the large variation in the number of drug pairs each side effect is associated with, there are only a limited number of drug pairs available for independently training models for prediction of different side effect types. As a result, polypharmacy side-effect prediction becomes a challenging task, especially when predicting rarer side effects, and thus it is important to develop an end-to-end approach such that the model is able to share information and learn from all side effects at once.

é¦–å…ˆï¼Œæˆ‘ä»¬è§‚å¯Ÿåˆ°åœ¨è¯ç‰©ç»„åˆä¸­æŸäº›å‰¯ä½œç”¨çš„å‘ç”Ÿé¢‘ç‡æœ‰å¾ˆå¤§çš„å·®å¼‚ã€‚æˆ‘ä»¬å‘ç°ï¼Œå·²çŸ¥è¶…è¿‡53%çš„å¤šè¯å‰¯ä½œç”¨å‘ç”Ÿåœ¨<3%çš„å·²è®°å½•è¯ç‰©ç»„åˆä¸­ï¼ˆå¦‚è„‘åŠ¨è„‰æ “å¡ã€è‚ºè„“è‚¿ã€è‚‰ç˜¤ã€èƒ¶åŸç´Šä¹±ï¼‰ã€‚ç›¸æ¯”ä¹‹ä¸‹ï¼Œæ›´é¢‘ç¹çš„å‰¯ä½œç”¨ï¼ˆå¦‚å‘•åã€ä½“é‡å¢åŠ ã€æ¶å¿ƒå’Œè´«è¡€ï¼‰å‘ç”Ÿé¢‘ç‡æ›´é«˜ã€‚ç”±äºä¸æ¯ç§å‰¯ä½œç”¨ç›¸å…³çš„è¯ç‰©å¯¹æ•°é‡å·®å¼‚å¾ˆå¤§ï¼Œå¯ç”¨äºé¢„æµ‹ä¸åŒå‰¯ä½œç”¨ç±»å‹çš„ç‹¬ç«‹è®­ç»ƒæ¨¡å‹çš„è¯ç‰©å¯¹æ•°é‡æœ‰é™ã€‚å› æ­¤ï¼Œå¤šè¯å‰¯ä½œç”¨é¢„æµ‹æˆä¸ºä¸€é¡¹å…·æœ‰æŒ‘æˆ˜æ€§çš„ä»»åŠ¡ï¼Œç‰¹åˆ«æ˜¯åœ¨é¢„æµ‹è¾ƒå°‘è§çš„å‰¯ä½œç”¨æ—¶ï¼Œå› æ­¤ï¼Œå¼€å‘ä¸€ç§ç«¯åˆ°ç«¯çš„æ–¹æ³•éå¸¸é‡è¦ï¼Œä»¥ä¾¿æ¨¡å‹èƒ½å¤Ÿå…±äº«ä¿¡æ¯å¹¶åŒæ—¶ä»æ‰€æœ‰å‰¯ä½œç”¨ä¸­å­¦ä¹ ã€‚

Second, we observe that polypharmacy side effects do not appear independently of one another in co-prescribed drug pairs (i.e. drug combinations), suggesting that joint modeling over multiple side effects can aid in the prediction task. To quantify the co-occurrence between side effects, we count the number of drug combinations in which a given side effect co-occurs with other side effects, and then use permutation testing with a null model of random co-occurrence. As exemplified for hypertension and nausea in Table 1, we find that the majority of the most common side effects are either significantly overrepresented or underrepresented with respect to how often they co-occur with nausea/hypertension as side effects in drug combinations, at Î±=0.05â . This observation points to the existence of mechanisms that may contribute to the shared pathophysiology of side effects, similar to what has been observed in disease comorbidity (Lee *et al.*, 2008). For example, we find that hypertension significantly co-occurs with anxiety but co-occurs less often with fever than dictated by random chance (Table 1). These relationships hold across the side effect dataset. We conclude that a prediction model should leverage dependence between side effects and be able to re-use the information learned about the molecular basis of one side effect to better understand the molecular basis of another side effect.

å…¶æ¬¡ï¼Œæˆ‘ä»¬è§‚å¯Ÿåˆ°ï¼Œåœ¨å…±åŒå¤„æ–¹çš„è¯ç‰©å¯¹ï¼ˆå³è¯ç‰©ç»„åˆï¼‰ä¸­ï¼Œå¤šè¯å‰¯ä½œç”¨ä¸ä¼šç›¸äº’ç‹¬ç«‹åœ°å‡ºç°ï¼Œè¿™è¡¨æ˜å¯¹å¤šç§å‰¯ä½œç”¨çš„è”åˆå»ºæ¨¡æœ‰åŠ©äºé¢„æµ‹ä»»åŠ¡ã€‚ä¸ºäº†é‡åŒ–å‰¯ä½œç”¨ä¹‹é—´çš„å…±åŒå‘ç”Ÿï¼Œæˆ‘ä»¬è®¡ç®—ç»™å®šå‰¯ä½œç”¨ä¸å…¶ä»–å‰¯ä½œç”¨å…±åŒå‘ç”Ÿçš„è¯ç‰©ç»„åˆçš„æ•°é‡ï¼Œç„¶åä½¿ç”¨éšæœºå…±åŒå‘ç”Ÿçš„ç©ºæ¨¡å‹è¿›è¡Œç½®æ¢æµ‹è¯•ã€‚æ­£å¦‚è¡¨ 1 ä¸­é«˜è¡€å‹å’Œæ¶å¿ƒçš„ä¾‹è¯ï¼Œæˆ‘ä»¬å‘ç°å¤§å¤šæ•°æœ€å¸¸è§çš„å‰¯ä½œç”¨åœ¨è¯ç‰©ç»„åˆä¸­ä½œä¸ºå‰¯ä½œç”¨ä¸æ¶å¿ƒ/é«˜è¡€å‹å…±åŒå‘ç”Ÿçš„é¢‘ç‡æ–¹é¢è¦ä¹ˆæ˜¾ç€è¿‡é«˜ï¼Œè¦ä¹ˆæ²¡æœ‰å……åˆ†ä½“ç°ï¼Œåœ¨Î±= 0.05æ—¶ã€‚è¿™ä¸€è§‚å¯Ÿç»“æœè¡¨æ˜å­˜åœ¨å¯èƒ½å¯¼è‡´å‰¯ä½œç”¨çš„å…±åŒç—…ç†ç”Ÿç†å­¦çš„æœºåˆ¶ï¼Œç±»ä¼¼äºåœ¨ç–¾ç—…åˆå¹¶ç—‡ä¸­è§‚å¯Ÿåˆ°çš„æƒ…å†µï¼ˆLee ç­‰ï¼Œ2008ï¼‰ã€‚ä¾‹å¦‚ï¼Œæˆ‘ä»¬å‘ç°é«˜è¡€å‹ä¸ç„¦è™‘æ˜¾ç€å…±åŒå‘ç”Ÿï¼Œä½†ä¸å‘çƒ§çš„å…±åŒå‘ç”Ÿé¢‘ç‡ä½äºéšæœºæœºä¼šï¼ˆè¡¨ 1ï¼‰ã€‚è¿™äº›å…³ç³»é€‚ç”¨äºå‰¯ä½œç”¨æ•°æ®é›†ã€‚æˆ‘ä»¬å¾—å‡ºçš„ç»“è®ºæ˜¯ï¼Œé¢„æµ‹æ¨¡å‹åº”è¯¥åˆ©ç”¨å‰¯ä½œç”¨ä¹‹é—´çš„ä¾èµ–å…³ç³»ï¼Œå¹¶èƒ½å¤Ÿé‡æ–°ä½¿ç”¨äº†è§£çš„å…³äºä¸€ç§å‰¯ä½œç”¨çš„åˆ†å­åŸºç¡€çš„ä¿¡æ¯ï¼Œä»¥æ›´å¥½åœ°äº†è§£å¦ä¸€ç§å‰¯ä½œç”¨çš„åˆ†å­åŸºç¡€ã€‚

**Table 1.**

Percent co-occurrence of hypertension and nausea with the 50 most frequent side effects in drug combinations, annotated with examples

è”åˆç”¨è¯ä¸­50ç§æœ€å¸¸è§å‰¯ä½œç”¨çš„é«˜è¡€å‹å’Œæ¶å¿ƒçš„å…±åŒå‘ç”Ÿç‡ï¼Œå¹¶é™„æœ‰å®ä¾‹è¯´æ˜

| Polypharmacy side effect *S *å¤šè¯å‰¯ä½œç”¨S | Overrepresented co-occurrence å¤šäººåŒæ—¶å‘ç”Ÿ                | Underrepresented co-occurrenceä»£è¡¨æ€§ä¸è¶³å…±åŒå‘ç”Ÿ           | **Insignificant co-occurrence***æ— æ„ä¹‰å…±åŒå‘ç”Ÿ** |
| ---------------------------------------- | --------------------------------------------------------- | ---------------------------------------------------------- | ------------------------------------------------ |
| Hypertensioné«˜è¡€å‹                       | 44% (hyperglycemia, anxiety, dizziness)é«˜è¡€ç³–ã€ç„¦è™‘ã€å¤´æ™• | 48% (fever, sepsis, dermatitis)å‘çƒ§ã€è´¥è¡€ç—‡ã€çš®ç‚          | 8% (cough, tachycardia)å’³å—½ã€å¿ƒåŠ¨è¿‡é€Ÿ            |
| Nauseaæ¶å¿ƒ                               | 54% (diarrhea, insomnia, asthenia)è…¹æ³»ã€å¤±çœ ã€ä¹åŠ›        | 34% (edema, anemia, neutropenia)æ°´è‚¿ã€è´«è¡€ã€ä¸­æ€§ç²’ç»†èƒå‡å°‘ | 12% (fever, dyspnea)å‘çƒ§ã€å‘¼å¸å›°éš¾               |

Note: The vast majority of side effects are either significantly overrepresented or underrepresented with respect to how often they appear in drug combinations with nausea/hypertension, at Î±=0.05
â , after Bonferroni correction.

æ³¨ï¼šç»å¤§å¤šæ•°å‰¯ä½œç”¨åœ¨ä¸æ¶å¿ƒ/é«˜è¡€å‹çš„è¯ç‰©ç»„åˆä¸­å‡ºç°çš„é¢‘ç‡æ˜æ˜¾è¿‡é«˜æˆ–è¿‡ä½ï¼ŒÎ±=0.05â ï¼Œç»è¿‡ Bonferroni ä¿®æ­£ã€‚

Third, we probe the relationship between proteins targeted by a drug pair and occurrence of side effects. Let *Ti* represent a set of target proteins associated with drug *i*, we then calculate the Jaccard similarity between target proteins of a given drug pair (*i*, *j*). We make several observations: (i) More than 68% of drug combinations have zero target proteins in common, suggesting it is important to use proteinâ€“protein interaction information to â€˜connectâ€™ different proteins targeted by different drugs. (ii) Random drug pairs have smaller overlap in targeted proteins than co-prescribed drugs (Fig. 2, light grey), *P*-valueâ€‰=â€‰5eâˆ’120â , 2-sample Kolmogorov-Smirnov (KS) test. (iii) We find that this trend is unequally observed across different side effects. For example, high blood pressure more strongly appears in drug combinations with shared target proteins than, for example, rib fracture (Fig. 2, purple). Over 150 side effects appear in combinations that differ significantly (at Î±=0.05 after Bonferroni correction) from the other true drug combinations, per a 2-sample KS test, suggesting a strong molecular basis of these side effects. Based on this findings, we conclude it is important for a model to consider how proteins interact with each other and to be able to model longer chains of (indirect) interactions.

ç¬¬ä¸‰ï¼Œæˆ‘ä»¬æ¢è®¨äº†è¯ç‰©å¯¹é¶å‘è›‹ç™½è´¨ä¸å‰¯ä½œç”¨å‘ç”Ÿä¹‹é—´çš„å…³ç³»ã€‚è®©*Ti*ä»£è¡¨ä¸€ç»„ä¸è¯ç‰©*i*ç›¸å…³çš„é¶è›‹ç™½ï¼Œç„¶åè®¡ç®—ç»™å®šè¯ç‰©å¯¹çš„é¶è›‹ç™½ä¹‹é—´çš„Jaccardç›¸ä¼¼æ€§ï¼ˆ*i*ï¼Œ*j*ï¼‰ã€‚æˆ‘ä»¬è¿›è¡Œäº†å‡ é¡¹è§‚å¯Ÿï¼šï¼ˆiï¼‰è¶…è¿‡ 68% çš„è¯ç‰©ç»„åˆçš„å…±åŒé¶è›‹ç™½ä¸ºé›¶ï¼Œè¿™è¡¨æ˜ä½¿ç”¨è›‹ç™½è´¨-è›‹ç™½è´¨ç›¸äº’ä½œç”¨ä¿¡æ¯â€œè¿æ¥â€ä¸åŒè¯ç‰©é¶å‘çš„ä¸åŒè›‹ç™½è´¨éå¸¸é‡è¦ã€‚ï¼ˆiiï¼‰éšæœºè¯ç‰©å¯¹åœ¨é¶å‘è›‹ç™½è´¨ä¸­çš„é‡å æ¯”è”åˆå¤„æ–¹è¯ç‰©å°ï¼ˆå›¾2ï¼Œæµ…ç°è‰²ï¼‰ï¼Œ*P*-å€¼â€‰=â€‰5eâˆ’120â , 2-æ ·æœ¬Kolmogorov-Smirnovï¼ˆKSï¼‰è¯•éªŒã€‚ï¼ˆiiiï¼‰æˆ‘ä»¬å‘ç°ï¼Œåœ¨ä¸åŒçš„å‰¯ä½œç”¨ä¸­ï¼Œè¿™ç§è¶‹åŠ¿æ˜¯ä¸å¹³ç­‰çš„ã€‚ä¾‹å¦‚ï¼Œä¸è‚‹éª¨éª¨æŠ˜ï¼ˆå›¾2ï¼Œç´«è‰²ï¼‰ç›¸æ¯”ï¼Œå…·æœ‰å…±åŒé¶è›‹ç™½çš„è¯ç‰©ç»„åˆæ›´å®¹æ˜“å‡ºç°é«˜è¡€å‹ã€‚æ ¹æ®2ä¸ªæ ·æœ¬çš„KSè¯•éªŒï¼Œè¶…è¿‡150ä¸ªå‰¯ä½œç”¨å‡ºç°åœ¨ä¸å…¶ä»–çœŸå®è¯ç‰©ç»„åˆæ˜¾è‘—ä¸åŒçš„ç»„åˆä¸­ï¼ˆBonferroniæ ¡æ­£åÎ±=0.05ï¼‰ï¼Œè¡¨æ˜è¿™äº›å‰¯ä½œç”¨å…·æœ‰å¼ºå¤§çš„åˆ†å­åŸºç¡€ã€‚åŸºäºè¿™ä¸€å‘ç°ï¼Œæˆ‘ä»¬å¾—å‡ºç»“è®ºï¼Œé‡è¦çš„æ˜¯ï¼Œæ¨¡å‹è€ƒè™‘è›‹ç™½è´¨å¦‚ä½•ç›¸äº’ä½œç”¨ï¼Œå¹¶èƒ½å¤Ÿå»ºæ¨¡æ›´é•¿çš„é“¾ï¼ˆé—´æ¥ï¼‰çš„ç›¸äº’ä½œç”¨ã€‚

![](index.assets/bty294f2.jpeg)

Jaccard similarity between target proteins for random pairs of drugs, all drug combinations and drug combinations associated with specific side effects. Drug pairs are stratified into three groups depending on whether drug i and j in a given pair (i, j) do not share any target proteins, share fewer than 50% target proteins, or share >50% target proteins (i.e. Jaccard(Ti,Tj)=0,â€‰0<Jaccard(Ti,Tj)<0.5 and 0.5â‰¤Jaccard(Ti,Tj)â‰¤1, respectively; Ti is a set of iâ€™s target proteins). We observe that drugs in most drug pairs, especially in random drug pairs (i.e. drugs not commonly co-prescribed, dark grey) have zero shared target proteins

éšæœºè¯ç‰©å¯¹ã€æ‰€æœ‰è¯ç‰©ç»„åˆå’Œä¸ç‰¹å®šå‰¯ä½œç”¨ç›¸å…³çš„è¯ç‰©ç»„åˆçš„é¶è›‹ç™½ä¹‹é—´çš„ç›¸ä¼¼æ€§ã€‚æ ¹æ®ç»™å®šè¯ç‰©å¯¹ï¼ˆiï¼Œjï¼‰ä¸­çš„è¯ç‰©iå’Œjæ˜¯å¦å…±äº«ä»»ä½•é¶è›‹ç™½ã€å…±äº«å°‘äº50%çš„é¶è›‹ç™½æˆ–å…±äº«>50%çš„é¶è›‹ç™½ï¼ˆå³Jaccardï¼ˆTiï¼ŒTjï¼‰=0ï¼Œâ€‰0<0.5å’Œ0.5â‰¤Jaccardï¼ˆTiï¼ŒTjï¼‰â‰¤åˆ†åˆ«ä¸º1ï¼›Tiæ˜¯içš„ä¸€ç»„é¶è›‹ç™½ï¼‰ã€‚æˆ‘ä»¬è§‚å¯Ÿåˆ°ï¼Œå¤§å¤šæ•°è¯ç‰©å¯¹ä¸­çš„è¯ç‰©ï¼Œå°¤å…¶æ˜¯éšæœºè¯ç‰©å¯¹ï¼ˆå³é€šå¸¸ä¸åˆç”¨çš„è¯ç‰©ï¼Œæ·±ç°è‰²ï¼‰æ²¡æœ‰å…±åŒçš„é¶è›‹ç™½

## 4 Graph convolutional *Decagon* approach

**å›¾å·ç§¯Decagonæ–¹æ³•**

We cast polypharmacy side effect modeling as a multirelational link prediction problem on a multimodal graph encoding drug, protein and side effect relationships (Fig. 1). More precisely, these relationships are represented by a graph G=(ğ’±,â„›)
 with N nodes (e.g. proteins, drugs) viâˆˆğ’± and labeled edges (relations) (vi,r,vj)â , where r is the edge type (relation type): (i) physical binding between two proteins, (ii) a target relationship between a drug and a protein or (iii) a particular type of a side effect between two drugs. As mentioned in Section 2, we consider 964 different relation types between drugs (i.e. side effects).

æˆ‘ä»¬å°†å¤šè¯å‰¯ä½œç”¨å»ºæ¨¡è§†ä¸ºç¼–ç è¯ç‰©ã€è›‹ç™½è´¨å’Œå‰¯ä½œç”¨å…³ç³»çš„å¤šæ¨¡å¼å›¾ä¸Šçš„å¤šå…³ç³»é“¾æ¥é¢„æµ‹é—®é¢˜ï¼ˆå›¾1ï¼‰ã€‚æ›´å‡†ç¡®åœ°è¯´ï¼Œè¿™äº›å…³ç³»ç”±å›¾Gè¡¨ç¤º=(ğ’±,â„›)å…·æœ‰Nä¸ªèŠ‚ç‚¹ï¼ˆä¾‹å¦‚è›‹ç™½è´¨ã€è¯ç‰©ï¼‰viâˆˆğ’±å’Œæ ‡è®°è¾¹ï¼ˆå…³ç³»ï¼‰ï¼ˆviï¼Œrï¼Œvjï¼‰â , å…¶ä¸­Ræ˜¯è¾¹ç¼˜å‹ï¼ˆå…³ç³»å‹ï¼‰ï¼šï¼ˆiï¼‰ä¸¤ç§è›‹ç™½è´¨ä¹‹é—´çš„ç‰©ç†ç»“åˆï¼Œï¼ˆiiï¼‰è¯ç‰©ä¸è›‹ç™½è´¨ä¹‹é—´çš„é¶å…³ç³»æˆ–ï¼ˆiiiï¼‰ä¸¤ç§è¯ç‰©ä¹‹é—´çš„å‰¯ä½œç”¨çš„ç‰¹å®šç±»å‹ã€‚å¦‚ç¬¬2èŠ‚æ‰€è¿°ï¼Œæˆ‘ä»¬è€ƒè™‘è¯ç‰©ä¹‹é—´çš„964ç§ä¸åŒçš„å…³ç³»ç±»å‹ï¼ˆå³å‰¯ä½œç”¨ï¼‰ã€‚

In addition, we allow for inclusion of side information in the form of additional node features. Different nodes (drugs, proteins) can have different number of node features, given by real-valued feature vectors x1,x2,â€¦,xN assigned to every node in the graph.

æ­¤å¤–ï¼Œæˆ‘ä»¬å…è®¸ä»¥é™„åŠ èŠ‚ç‚¹ç‰¹å¾çš„å½¢å¼åŒ…å«è¾¹ä¿¡æ¯ã€‚ä¸åŒçš„èŠ‚ç‚¹ï¼ˆè¯ç‰©ã€è›‹ç™½è´¨ï¼‰å¯ä»¥å…·æœ‰ä¸åŒæ•°é‡çš„èŠ‚ç‚¹ç‰¹å¾ï¼Œç”±å®å€¼ç‰¹å¾å‘é‡x1ã€x2ã€â€¦ã€xNç»™å‡ºåˆ†é…ç»™å›¾å½¢ä¸­çš„æ¯ä¸ªèŠ‚ç‚¹ã€‚

Polypharmacy side effect prediction task. The polypharmacy side effect prediction task considers the problem of identifying associations between drug pairs and side effects. Importantly, these associations are limited to only those that cannot be attributed to either drug alone. Using the graph G, the task is to predict labeled edges between drug nodes. Given a drug pair (vi, vj), our aim is to determine how likely an edge eij=(vi,r,vj) of type r belongs to â„›â , meaning that concurrent use of drugs vi and vj [i.e. the use of a drug combination (vi, vj)] is associated with a polypharmacy side effect of type r in the human patient population.

å¤šè¯å‰¯ä½œç”¨é¢„æµ‹ä»»åŠ¡ã€‚å¤šè¯å‰¯ä½œç”¨é¢„æµ‹ä»»åŠ¡è€ƒè™‘è¯†åˆ«è¯ç‰©å¯¹å’Œå‰¯ä½œç”¨ä¹‹é—´çš„å…³è”çš„é—®é¢˜ã€‚é‡è¦çš„æ˜¯ï¼Œè¿™äº›å…³è”ä»…é™äºé‚£äº›ä¸èƒ½å•ç‹¬å½’å› äºä¸¤ç§è¯ç‰©çš„å…³è”ã€‚ä½¿ç”¨å›¾Gï¼Œä»»åŠ¡æ˜¯é¢„æµ‹è¯ç‰©èŠ‚ç‚¹ä¹‹é—´çš„æ ‡è®°è¾¹ã€‚ç»™å®šä¸€å¯¹è¯ç‰©ï¼ˆviï¼Œvjï¼‰ï¼Œæˆ‘ä»¬çš„ç›®æ ‡æ˜¯ç¡®å®šrå‹è¾¹ç¼˜eij=ï¼ˆviï¼Œrï¼Œvjï¼‰å±äºâ„›â , è¿™æ„å‘³ç€åŒæ—¶ä½¿ç”¨è¯ç‰©viå’Œvj[å³ä½¿ç”¨è¯ç‰©ç»„åˆï¼ˆviï¼Œvjï¼‰]ä¸äººç±»æ‚£è€…ç¾¤ä½“ä¸­rå‹å¤šè¯å‰¯ä½œç”¨æœ‰å…³ã€‚

To this aim, we develop a non-linear, multi-layer convolutional graph neural network model Decagon that operates directly on graph G. Decagon has two main components:

ä¸ºæ­¤ï¼Œæˆ‘ä»¬å¼€å‘äº†ä¸€ç§ç›´æ¥ä½œç”¨äºå›¾Gçš„éçº¿æ€§å¤šå±‚å·ç§¯å›¾ç¥ç»ç½‘ç»œæ¨¡å‹Decagonã€‚Decagonæœ‰ä¸¤ä¸ªä¸»è¦ç»„æˆéƒ¨åˆ†ï¼š

an encoder: a graph convolutional network operating on G and producing embeddings for nodes in G (Fig. 3A; Section 4.1) and
a decoder: a tensor factorization model using these embeddings to model polypharmacy side effects (Fig. 3B; Section 4.2).

ç¼–ç å™¨ï¼šåœ¨Gä¸Šè¿è¡Œå¹¶ä¸ºGä¸­çš„èŠ‚ç‚¹ç”ŸæˆåµŒå…¥çš„å›¾å·ç§¯ç½‘ç»œï¼ˆå›¾3Aï¼›ç¬¬4.1èŠ‚ï¼‰å’Œè§£ç å™¨ï¼šä½¿ç”¨è¿™äº›åµŒå…¥å»ºæ¨¡å¤šè¯å‰¯ä½œç”¨çš„å¼ é‡åˆ†è§£æ¨¡å‹ï¼ˆå›¾3Bï¼›ç¬¬4.2èŠ‚ï¼‰ã€‚

![](index.assets/bty294f3.jpeg)

Overview of Decagon model architecture. (A) An Decagon encoder. Shown is a per-layer update for a single graph node (a drug node representing Ciprofloxacin based on the small example input graph in Fig. 1). Hidden state activations from neighboring nodes Nrc are gathered and then transformed for each relation type r individually (i.e. gastrointestinal bleed, bradycardia and drug target relation). The resulting representation is accumulated in a (normalized) sum and passed through a non-linear activation function (i.e. ReLU) to produce hidden state of node vc in the (k+1)-th layer, hc(k+1). This per-node update is computed in parallel with shared parameters across the whole graph. (B) For every relation, Decagon decoder takes pairs of embeddings (e.g. hidden node representations zc and zs representing Ciprofloxacin and Simvastatin) and produces a score for every (potential) edge in the graph. Shown is the decoder for poypharmacy side effects relation types. (C) A batch of neural networks that compute embeddings of six drug nodes in the input graph. In Decagon, neural networks differ from node to node but they all share the same set of relation-specific trainable parameters [i.e. the parameters of the encoder and decoder; see Equations (1) and (2)]. That is, rectangles with the same shading patterns share parameters, and thin rectangles with black and white shading pattern denote densely connected neural layers

Decagonæ¨¡å‹ä½“ç³»ç»“æ„æ¦‚è¿°ã€‚ï¼ˆAï¼‰ Decagonç¼–ç å™¨ã€‚æ‰€ç¤ºä¸ºå•ä¸ªå›¾èŠ‚ç‚¹çš„æ¯å±‚æ›´æ–°ï¼ˆåŸºäºå›¾1ä¸­çš„å°ç¤ºä¾‹è¾“å…¥å›¾ï¼Œä»£è¡¨ç¯ä¸™æ²™æ˜Ÿçš„è¯ç‰©èŠ‚ç‚¹ï¼‰ã€‚æ”¶é›†ç›¸é‚»èŠ‚ç‚¹Nrcçš„éšè—çŠ¶æ€æ¿€æ´»ï¼Œç„¶ååˆ†åˆ«è½¬æ¢ä¸ºæ¯ç§å…³ç³»ç±»å‹rï¼ˆå³èƒƒè‚ é“å‡ºè¡€ã€å¿ƒåŠ¨è¿‡ç¼“å’Œè¯ç‰©é¶ç‚¹å…³ç³»ï¼‰ã€‚ç»“æœè¡¨ç¤ºç´¯ç§¯åœ¨ï¼ˆå½’ä¸€åŒ–ï¼‰å’Œä¸­ï¼Œå¹¶é€šè¿‡éçº¿æ€§æ¿€æ´»å‡½æ•°ï¼ˆå³ReLUï¼‰æ¥äº§ç”Ÿç¬¬ï¼ˆk+1ï¼‰å±‚ä¸­èŠ‚ç‚¹vcçš„éšè—çŠ¶æ€hcï¼ˆk+1ï¼‰ã€‚æ­¤æ¯èŠ‚ç‚¹æ›´æ–°ä¸æ•´ä¸ªå›¾ä¸­çš„å…±äº«å‚æ•°å¹¶è¡Œè®¡ç®—ã€‚ï¼ˆBï¼‰ å¯¹äºæ¯ä¸ªå…³ç³»ï¼ŒDecagonè§£ç å™¨é‡‡ç”¨æˆå¯¹åµŒå…¥ï¼ˆä¾‹å¦‚ï¼Œéšè—èŠ‚ç‚¹è¡¨ç¤ºzcå’Œzsè¡¨ç¤ºç¯ä¸™æ²™æ˜Ÿå’Œè¾›ä¼ä»–æ±€ï¼‰ï¼Œå¹¶ä¸ºå›¾ä¸­çš„æ¯ä¸ªï¼ˆæ½œåœ¨ï¼‰è¾¹ç”Ÿæˆåˆ†æ•°ã€‚æ‰€ç¤ºä¸ºpoypharmacyå‰¯ä½œç”¨å…³ç³»ç±»å‹çš„è§£ç å™¨ã€‚ï¼ˆCï¼‰ è®¡ç®—è¾“å…¥å›¾ä¸­å…­ä¸ªè¯ç‰©èŠ‚ç‚¹åµŒå…¥çš„ä¸€æ‰¹ç¥ç»ç½‘ç»œã€‚åœ¨Decagonä¸­ï¼Œç¥ç»ç½‘ç»œå› èŠ‚ç‚¹è€Œå¼‚ï¼Œä½†å®ƒä»¬éƒ½å…·æœ‰ç›¸åŒçš„ä¸€ç»„ç‰¹å®šäºå…³ç³»çš„å¯è®­ç»ƒå‚æ•°[å³ç¼–ç å™¨å’Œè§£ç å™¨çš„å‚æ•°ï¼›è§ç­‰å¼ï¼ˆ1ï¼‰å’Œï¼ˆ2ï¼‰]ã€‚ä¹Ÿå°±æ˜¯è¯´ï¼Œå…·æœ‰ç›¸åŒç€è‰²æ¨¡å¼çš„çŸ©å½¢å…±äº«å‚æ•°ï¼Œè€Œå…·æœ‰é»‘ç™½ç€è‰²æ¨¡å¼çš„è–„çŸ©å½¢è¡¨ç¤ºå¯†é›†è¿æ¥çš„ç¥ç»å±‚ã€‚

### 4.1 Graph convolutional encoder

**å›¾å·ç§¯ç¼–ç å™¨**

We first describe the graph encoder model, which takes as input a graph *G* and additional node feature vectors **x**iâ , and produces a node *d*-dimensional embedding **z**iâˆˆâ„d for every node (drug, protein) in the graph.

æˆ‘ä»¬é¦–å…ˆæè¿°äº†å›¾å½¢ç¼–ç å™¨æ¨¡å‹ï¼Œè¯¥æ¨¡å‹ä»¥å›¾å½¢*G*å’Œå…¶ä»–èŠ‚ç‚¹ç‰¹å¾å‘é‡**x**iä½œä¸ºè¾“å…¥â , å¹¶ç”Ÿæˆä¸€ä¸ªèŠ‚ç‚¹*d*-ç»´åµŒå…¥**z**iâˆˆâ„dè¡¨ç¤ºå›¾ä¸­çš„æ¯ä¸ªèŠ‚ç‚¹ï¼ˆè¯ç‰©ã€è›‹ç™½è´¨ï¼‰ã€‚

We propose an encoder model that makes efficient use of information sharing across regions in the graph and assigns separate processing channels for each relation type. The idea is that *Decagon* learns how to transform and propagate information, captured by node feature vectors, across the graph. Every nodeâ€™s network neighborhood defines a different neural network information propagation architecture but these architectures then share functions/parameters that define how information is shared and propagated. We learn convolutional operators that propagate and transform information across different parts of the graph and across different relation types. The model inspired by a recent class of convolutional neural networks that operate directly on graphs (Defferrard *et al.*, 2016; Kipf and Welling, 2016). For a given node *Decagon* performs transformation/aggregation operations on feature vectors of its neighbors. This way *Decagon* only takes into account the first-order neighborhood of a node and applies the same transformation across all locations in the graph. Successive application of these operations then effectively convolves information across the *K*-th order neighborhood (i.e. embedding of a node depends on all the nodes that are at most *K* steps away), where *K* is the number of successive operations of convolutional layers in the neural network model.

æˆ‘ä»¬æå‡ºäº†ä¸€ä¸ªç¼–ç å™¨æ¨¡å‹ï¼Œè¯¥æ¨¡å‹æœ‰æ•ˆåœ°åˆ©ç”¨äº†å›¾ä¸­è·¨åŒºåŸŸçš„ä¿¡æ¯å…±äº«ï¼Œå¹¶ä¸ºæ¯ç§å…³ç³»ç±»å‹åˆ†é…äº†å•ç‹¬çš„å¤„ç†é€šé“ã€‚å…¶æ€æƒ³æ˜¯ï¼Œ*Decagon*å­¦ä¹ å¦‚ä½•åœ¨å›¾å½¢ä¸­è½¬æ¢å’Œä¼ æ’­ç”±èŠ‚ç‚¹ç‰¹å¾å‘é‡æ•è·çš„ä¿¡æ¯ã€‚æ¯ä¸ªèŠ‚ç‚¹çš„ç½‘ç»œé‚»åŸŸå®šä¹‰äº†ä¸åŒçš„ç¥ç»ç½‘ç»œä¿¡æ¯ä¼ æ’­ä½“ç³»ç»“æ„ï¼Œä½†è¿™äº›ä½“ç³»ç»“æ„å…±äº«å®šä¹‰ä¿¡æ¯å…±äº«å’Œä¼ æ’­æ–¹å¼çš„å‡½æ•°/å‚æ•°ã€‚æˆ‘ä»¬å­¦ä¹ å·ç§¯è¿ç®—ç¬¦ï¼Œå®ƒåœ¨å›¾çš„ä¸åŒéƒ¨åˆ†å’Œä¸åŒå…³ç³»ç±»å‹ä¹‹é—´ä¼ æ’­å’Œè½¬æ¢ä¿¡æ¯ã€‚è¯¥æ¨¡å‹å—æœ€è¿‘ä¸€ç±»ç›´æ¥åœ¨å›¾å½¢ä¸Šè¿è¡Œçš„å·ç§¯ç¥ç»ç½‘ç»œçš„å¯å‘ï¼ˆDefferard*ç­‰ï¼Œ2016ï¼›Kipfå’ŒWellingï¼Œ2016ï¼‰ã€‚å¯¹äºç»™å®šèŠ‚ç‚¹ï¼ŒDecagon*å¯¹å…¶ç›¸é‚»èŠ‚ç‚¹çš„ç‰¹å¾å‘é‡æ‰§è¡Œå˜æ¢/èšåˆæ“ä½œã€‚è¿™æ ·ï¼Œ*Decagon*åªè€ƒè™‘èŠ‚ç‚¹çš„ä¸€é˜¶é‚»åŸŸï¼Œå¹¶åœ¨å›¾ä¸­çš„æ‰€æœ‰ä½ç½®åº”ç”¨ç›¸åŒçš„å˜æ¢ã€‚éšåï¼Œè¿™äº›æ“ä½œçš„è¿ç»­åº”ç”¨æœ‰æ•ˆåœ°åœ¨*K*é˜¶é‚»åŸŸä¸Šå·ç§¯ä¿¡æ¯ï¼ˆå³ï¼ŒèŠ‚ç‚¹çš„åµŒå…¥å–å†³äºæœ€å¤š*K*æ­¥çš„æ‰€æœ‰èŠ‚ç‚¹ï¼‰ï¼Œå…¶ä¸­*K*æ˜¯ç¥ç»ç½‘ç»œæ¨¡å‹ä¸­å·ç§¯å±‚çš„è¿ç»­æ“ä½œæ•°ã€‚

In each layer, *Decagon* propagates latent node feature information across edges of the graph, while taking into account the type (relation) of an edge (Schlichtkrull *et al.*, 2017). A single layer of this neural network model takes the following form:

åœ¨æ¯ä¸€å±‚ä¸­ï¼Œ*Decagon*å°†æ½œåœ¨èŠ‚ç‚¹ç‰¹å¾ä¿¡æ¯ä¼ æ’­åˆ°å›¾å½¢çš„è¾¹ç¼˜ï¼ŒåŒæ—¶è€ƒè™‘è¾¹ç¼˜çš„ç±»å‹ï¼ˆå…³ç³»ï¼‰ï¼ˆSchlichtkrull*ç­‰äººï¼Œ2017å¹´ï¼‰ã€‚è¯¥ç¥ç»ç½‘ç»œæ¨¡å‹çš„å•å±‚é‡‡ç”¨ä»¥ä¸‹å½¢å¼ï¼š

![image-20210929163020256](index.assets/image-20210929163020256.png)

A deeper model can be built by chaining multiple (i.e. *K*) of these layers (Fig. 3A) with appropriate activation functions. To arrive at the final embedding **z**iâˆˆâ„d of node *vi*, we compute its representation as: **z**i=**h**(K)i. The overall encoder then takes the following form. We stack *K* layers as defined in [Equation (1)](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E1) such that the output of the previous layer becomes the input to the next layer. The input to the first layer are node feature vectors, **h**(0)i=**x**iâ , or unique one-hot vectors for every node in the graph if no features are present.

é€šè¿‡å°†è¿™äº›å±‚ï¼ˆå›¾3Aï¼‰ä¸­çš„å¤šä¸ªå±‚ï¼ˆå³*K*ï¼‰ä¸é€‚å½“çš„æ¿€æ´»å‡½æ•°é“¾æ¥ï¼Œå¯ä»¥å»ºç«‹æ›´æ·±å±‚çš„æ¨¡å‹ã€‚è¦è¾¾åˆ°æœ€ç»ˆåµŒå…¥**z**iâˆˆâ„å¯¹äºèŠ‚ç‚¹*vi*ï¼Œæˆ‘ä»¬è®¡ç®—å…¶è¡¨ç¤ºä¸ºï¼š**z**i=**h**ï¼ˆKï¼‰iã€‚ç„¶åï¼Œæ•´ä¸ªç¼–ç å™¨é‡‡ç”¨ä»¥ä¸‹å½¢å¼ã€‚æˆ‘ä»¬æŒ‰ç…§[å…¬å¼ï¼ˆ1ï¼‰]ä¸­çš„å®šä¹‰å †å *K*å±‚(https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E1)ä½¿å‰ä¸€å±‚çš„è¾“å‡ºæˆä¸ºä¸‹ä¸€å±‚çš„è¾“å…¥ã€‚ç¬¬ä¸€å±‚çš„è¾“å…¥æ˜¯èŠ‚ç‚¹ç‰¹å¾å‘é‡ï¼Œ**h**ï¼ˆ0ï¼‰i=**x**iâ , æˆ–è€…ï¼Œå¦‚æœä¸å­˜åœ¨ä»»ä½•ç‰¹å¾ï¼Œåˆ™ä¸ºå›¾ä¸­çš„æ¯ä¸ªèŠ‚ç‚¹æä¾›ä¸€ä¸ªå”¯ä¸€çš„çƒ­å‘é‡ã€‚

### 4.2 Tensor factorization decoder

**å¼ é‡åˆ†è§£è¯‘ç å™¨**

So far, we introduced *Decagon*â€™s encoder. The encoder maps each node viâˆˆğ’± to a an embedding, a real-valued vector representation **z**iâˆˆâ„dâ , where *d* is the dimensionality of node representations. We proceed by describing the decoder component of *Decagon*.

åˆ°ç›®å‰ä¸ºæ­¢ï¼Œæˆ‘ä»¬ä»‹ç»äº†*Decagon*çš„ç¼–ç å™¨ã€‚ç¼–ç å™¨æ˜ å°„æ¯ä¸ªèŠ‚ç‚¹viâˆˆğ’± å¯¹äºåµŒå…¥ï¼Œå®å€¼å‘é‡è¡¨ç¤º**z**iâˆˆâ„Dâ , å…¶ä¸­*d*æ˜¯èŠ‚ç‚¹è¡¨ç¤ºçš„ç»´åº¦ã€‚æˆ‘ä»¬ç»§ç»­æè¿°*Decagon*çš„è§£ç å™¨ç»„ä»¶ã€‚

The goal of decoder is to reconstruct labeled edges in *G* by relying on learned node embeddings and by treating each label (edge type) differently. In particular, decoder scores a (vi,r,vj)-triple through a function *g* whose goal is to assign a score g(vi,r,vj) representing how likely it is that drugs *vi* and *vj* are interacting through a relation/side effect type *r* (Fig. 3B). Using embeddings for nodes *i* and *j* returned by *Decagonâ€™*s encoder (Section 4.1)

è§£ç å™¨çš„ç›®æ ‡æ˜¯é€šè¿‡ä¾èµ–å­¦ä¹ çš„èŠ‚ç‚¹åµŒå…¥å’Œå¯¹æ¯ä¸ªæ ‡ç­¾ï¼ˆè¾¹ç¼˜ç±»å‹ï¼‰çš„ä¸åŒå¤„ç†æ¥é‡å»º*G*ä¸­çš„æ ‡è®°è¾¹ç¼˜ã€‚å…·ä½“è€Œè¨€ï¼Œè§£ç å™¨é€šè¿‡å‡½æ•°*g*å¯¹aï¼ˆviï¼Œrï¼Œvjï¼‰-ä¸‰å…ƒç»„è¿›è¡Œè¯„åˆ†ï¼Œè¯¥å‡½æ•°çš„ç›®æ ‡æ˜¯åˆ†é…åˆ†æ•°gï¼ˆviï¼Œrï¼Œvjï¼‰ï¼Œè¯¥åˆ†æ•°è¡¨ç¤ºè¯ç‰©*vi*å’Œ*vj*é€šè¿‡å…³ç³»/å‰¯ä½œç”¨ç±»å‹*r*ç›¸äº’ä½œç”¨çš„å¯èƒ½æ€§ï¼ˆå›¾3Bï¼‰ã€‚å¯¹*Decagon'*sç¼–ç å™¨è¿”å›çš„èŠ‚ç‚¹*i*å’Œ*j*ä½¿ç”¨åµŒå…¥ï¼ˆç¬¬4.1èŠ‚ï¼‰

![image-20210929163314686](index.assets/image-20210929163314686.png)

Next, we explain *Decagonâ€™*s decoder by distinguishing between the following two cases: 

æ¥ä¸‹æ¥ï¼Œæˆ‘ä»¬é€šè¿‡åŒºåˆ†ä»¥ä¸‹ä¸¤ç§æƒ…å†µæ¥è§£é‡Š*Decagon'*sè§£ç å™¨ï¼š

1.  When *vi* and *vj* are drug nodes, the decoder *g* in [Equation (2)](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E2) assumes a global model of drugâ€“drug interactions (i.e. **R**) whose variation and importance across polypharmacy side effects are described by side-effect-specific diagonal factors (i.e. **D**râ ). Here, **R** is a trainable parameter matrix of shape *d***Ã—***d* that models global drugâ€“drug interactions across all possible polypharmacy side effects. Additionally, in *Decagon*, every relation *r* representing a different polypharmacy side effect is associated with a diagonal *d***Ã—***d* matrix **D**r modeling the importance of each dimension in **z**i towards side effect *r*. In an alternative view, this decoder can be thought of as a tensor factorization [more specifically, a rank-*d* DEDICOM tensor decomposition (Nickel *et al.*, 2011; Trouillon *et al.*, 2016)] of a three-way tensor, where two modes are identically formed by the drugs and the third mode holds polypharmacy side effects of drug combinations. However, a distinguishing characteristic of *Decagon* is the reliance on the encoder. Whereas classic tensor decompositions use node representations optimized directly in training, we compute them in an end-to-end fashion where node embeddings are optimized jointly together with the tensor factorization.

    å½“*vi*å’Œ*vj*æ˜¯è¯ç‰©èŠ‚ç‚¹æ—¶ï¼Œè§£ç å™¨*g*åœ¨[ç­‰å¼ï¼ˆ2ï¼‰](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E2)å‡è®¾è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨çš„å…¨å±€æ¨¡å‹ï¼ˆå³.*R**ï¼‰ï¼Œå…¶åœ¨å¤šè¯å‰¯ä½œç”¨ä¸­çš„å˜åŒ–å’Œé‡è¦æ€§ç”±å‰¯ä½œç”¨ç‰¹å®šçš„å¯¹è§’å› å­ï¼ˆå³.*D**Rï¼‰æè¿°â ). åœ¨è¿™é‡Œï¼Œ**R**æ˜¯ä¸€ä¸ªå½¢çŠ¶ä¸º*d*********d*çš„å¯è®­ç»ƒå‚æ•°çŸ©é˜µï¼Œå®ƒæ¨¡æ‹Ÿäº†æ‰€æœ‰å¯èƒ½çš„å¤šè¯å‰¯ä½œç”¨çš„å…¨çƒè¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ã€‚æ­¤å¤–ï¼Œåœ¨*Decagon*ä¸­ï¼Œä»£è¡¨ä¸åŒå¤šè¯å‰¯ä½œç”¨çš„æ¯ä¸ªå…³ç³»*R*éƒ½ä¸å¯¹è§’*d*********d*çŸ©é˜µ**d**Rå…³è”ï¼Œè¯¥çŸ©é˜µæ¨¡æ‹Ÿäº†æ¯ä¸ªäºŒèšä½“çš„é‡è¦æ€§åœ¨å¦ä¸€ç§è§‚ç‚¹ä¸­ï¼Œè¯¥è§£ç å™¨å¯è¢«è§†ä¸ºå¼ é‡åˆ†è§£[æ›´å…·ä½“åœ°è¯´ï¼Œç§©-*d*DEDICOMå¼ é‡åˆ†è§£ï¼ˆNickel*ç­‰äººï¼Œ2011ï¼›Trouillon*ç­‰äººï¼Œ2016ï¼‰]å¯¹äºä¸‰å‘å¼ é‡ï¼Œå…¶ä¸­ä¸¤ä¸ªæ¨¡å¼ç”±è¯ç‰©ç›¸åŒå½¢æˆï¼Œç¬¬ä¸‰ä¸ªæ¨¡å¼å…·æœ‰è¯ç‰©ç»„åˆçš„å¤šè¯å‰¯ä½œç”¨ã€‚ç„¶è€Œï¼Œ*Decagon*çš„ä¸€ä¸ªæ˜¾è‘—ç‰¹å¾æ˜¯ä¾èµ–ç¼–ç å™¨ã€‚è€Œç»å…¸å¼ é‡åˆ†è§£ä½¿ç”¨åœ¨è®­ç»ƒä¸­ç›´æ¥ä¼˜åŒ–çš„èŠ‚ç‚¹è¡¨ç¤ºï¼Œæˆ‘ä»¬åœ¨ä¸€ç§ç«¯åˆ°ç«¯çš„æ–¹å¼ï¼Œå…¶ä¸­èŠ‚ç‚¹åµŒå…¥ä¸å¼ é‡åˆ†è§£ä¸€èµ·ä¼˜åŒ–ã€‚

2.  When *vi* and *vj* are not both drug nodes, the decoder *g* in [Equation (2)](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E2) employs a bilinear form to decode edges from node embeddings. More precisely, in that case, the decoding function *g* is associated with a trainable parameter matrix **M**r of shape *d***Ã—***d* that models interactions between every two dimensions in **z**i and **z**jâ . The predicted edge probability is then computed using a bilinear form *(*[Equation 2](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E2)) followed by the application of a sigmoid function *(*[Equation 3](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E3)).

    å½“*vi*å’Œ*vj*ä¸æ˜¯ä¸¤ä¸ªè¯ç‰©èŠ‚ç‚¹æ—¶ï¼Œè§£ç å™¨*g*é‡‡ç”¨åŒçº¿æ€§å½¢å¼å¯¹èŠ‚ç‚¹åµŒå…¥çš„è¾¹è¿›è¡Œè§£ç ã€‚æ›´å‡†ç¡®åœ°è¯´ï¼Œåœ¨è¿™ç§æƒ…å†µä¸‹ï¼Œè§£ç å‡½æ•°*g*ä¸å½¢çŠ¶ä¸º*d***Ã—****d*çš„å¯è®­ç»ƒå‚æ•°çŸ©é˜µ**M**rç›¸å…³è”ï¼Œè¯¥çŸ©é˜µå¯¹**z**iå’Œ**z**jä¸­æ¯ä¸¤ä¸ªç»´åº¦ä¹‹é—´çš„äº¤äº’è¿›è¡Œå»ºæ¨¡â . ç„¶åä½¿ç”¨åŒçº¿æ€§å½¢å¼*ï¼ˆ*[ç­‰å¼2]è®¡ç®—é¢„æµ‹çš„è¾¹ç¼˜æ¦‚ç‡ç„¶ååº”ç”¨Så½¢å‡½æ•°.

### 4.3 *Decagon* model training

**decagonæ¨¡å‹è®­ç»ƒ**

During model training, we optimize model parameters using the cross-entropy loss:

åœ¨æ¨¡å‹è®­ç»ƒæœŸé—´ï¼Œæˆ‘ä»¬ä½¿ç”¨äº¤å‰ç†µæŸå¤±ä¼˜åŒ–æ¨¡å‹å‚æ•°ï¼š

![image-20210929163358130](index.assets/image-20210929163358130.png)

Recent results have shown that modeling graph-structured data can often be significantly improved with end-to-end learning (Defferrard *et al.*, 2016; Gilmer *et al.*, 2017), thus we take an end-to-end optimization approach and jointly optimize over all trainable parameters and propagate loss function gradients through both *Decagonâ€™*s encoder as well as decoder.

æœ€è¿‘çš„ç ”ç©¶ç»“æœè¡¨æ˜ï¼Œé€šè¿‡ç«¯åˆ°ç«¯å­¦ä¹ ï¼Œå›¾å½¢ç»“æ„åŒ–æ•°æ®çš„å»ºæ¨¡é€šå¸¸å¯ä»¥å¾—åˆ°æ˜¾è‘—æ”¹è¿›ï¼ˆDefferard*ç­‰ï¼Œ2016å¹´ï¼›Gilmer*ç­‰ï¼Œ2017å¹´ï¼‰ï¼Œå› æ­¤ï¼Œæˆ‘ä»¬é‡‡ç”¨ç«¯åˆ°ç«¯ä¼˜åŒ–æ–¹æ³•ï¼Œå¯¹æ‰€æœ‰å¯è®­ç»ƒå‚æ•°è¿›è¡Œè”åˆä¼˜åŒ–ï¼Œå¹¶é€šè¿‡*Decagon'*sç¼–ç å™¨å’Œè§£ç å™¨ä¼ æ’­æŸå¤±å‡½æ•°æ¢¯åº¦ã€‚

To optimize the model we train it for a maximum of 100 epochs (training iterations) using the Adam optimizer (Kingma and Ba, 2014) with a learning rate of 0.001 and early stopping with a window size of 2, i.e. we stop training if the validation loss does not decrease for two consecutive epochs. We initialize weights using the initialization described in Glorot and Bengio (2010) and accordingly normalize node feature vectors. In order for the model to generalize well to unobserved edges we apply a regular dropout (Srivastava *et al.*, 2014) to hidden layer units ([Equation 1](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E1)). In practice, we use efficient sparse matrix multiplications, with complexity linear in the number of edges in *G*, to implement *Decagon* model.

ä¸ºäº†ä¼˜åŒ–æ¨¡å‹ï¼Œæˆ‘ä»¬ä½¿ç”¨Adamä¼˜åŒ–å™¨ï¼ˆKingmaå’ŒBaï¼Œ2014ï¼‰ä»¥0.001çš„å­¦ä¹ ç‡å¯¹æ¨¡å‹è¿›è¡Œæœ€å¤š100ä¸ªepochsçš„è®­ç»ƒï¼ˆè®­ç»ƒè¿­ä»£ï¼‰ï¼Œå¹¶æå‰åœæ­¢ï¼Œçª—å£å¤§å°ä¸º2ï¼Œå³å¦‚æœè¿ç»­ä¸¤ä¸ªepochsçš„éªŒè¯æŸå¤±æ²¡æœ‰å‡å°‘ï¼Œæˆ‘ä»¬åœæ­¢è®­ç»ƒã€‚æˆ‘ä»¬ä½¿ç”¨Glorotå’ŒBengioï¼ˆ2010ï¼‰ä¸­æè¿°çš„åˆå§‹åŒ–åˆå§‹åŒ–æƒé‡ï¼Œå¹¶ç›¸åº”åœ°è§„èŒƒåŒ–èŠ‚ç‚¹ç‰¹å¾å‘é‡ã€‚ä¸ºäº†ä½¿æ¨¡å‹èƒ½å¤Ÿå¾ˆå¥½åœ°æ¨å¹¿åˆ°æœªè§‚æµ‹åˆ°çš„è¾¹ç¼˜ï¼Œæˆ‘ä»¬å¯¹éšè—å±‚å•å…ƒï¼ˆæ–¹ç¨‹å¼1ï¼‰åº”ç”¨äº†ä¸€ä¸ªè§„åˆ™çš„è¡°å‡. åœ¨å®è·µä¸­ï¼Œæˆ‘ä»¬ä½¿ç”¨æœ‰æ•ˆçš„ç¨€ç–çŸ©é˜µä¹˜æ³•æ¥å®ç°*Decagon*æ¨¡å‹ï¼Œå…¶å¤æ‚åº¦ä¸*G*ä¸­çš„è¾¹æ•°æˆçº¿æ€§å…³ç³»ã€‚

We use mini-batching by sampling contributions to the loss function in [Equation (5)](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E5). That is, we process multiple training mini-batches, each obtained by sampling only a fixed number of contributions from the sum over edges in [Equation (5)](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E5), resulting in dynamic batches of computation graphs (Fig. 3C). By only considering a fixed number of contributions to the loss function, we can remove respective data points that do not appear in the current mini-batch. This serves as an effective means of regularization, and reduces the memory requirement to train the model, which is necessary so that we can fit the full model into GPU memory (all data and code are released on the project website).

æˆ‘ä»¬é€šè¿‡å¯¹[ç­‰å¼ï¼ˆ5ï¼‰]ä¸­æŸå¤±å‡½æ•°çš„æŠ½æ ·è´¡çŒ®ä½¿ç”¨å°æ‰¹é‡ã€‚ä¹Ÿå°±æ˜¯è¯´ï¼Œæˆ‘ä»¬å¤„ç†å¤šä¸ªè®­ç»ƒå°æ‰¹é‡ï¼Œæ¯ä¸ªå°æ‰¹é‡éƒ½æ˜¯é€šè¿‡ä»[ç­‰å¼ï¼ˆ5ï¼‰]ä¸­çš„è¾¹å’Œä¸­åªæŠ½æ ·å›ºå®šæ•°é‡çš„è´¡çŒ®è€Œè·å¾—çš„ï¼Œç”ŸæˆåŠ¨æ€æ‰¹é‡è®¡ç®—å›¾ï¼ˆå›¾3Cï¼‰ã€‚é€šè¿‡åªè€ƒè™‘æŸå¤±å‡½æ•°çš„å›ºå®šè´¡çŒ®æ•°ï¼Œæˆ‘ä»¬å¯ä»¥åˆ é™¤å½“å‰å°æ‰¹é‡ä¸­æœªå‡ºç°çš„å„ä¸ªæ•°æ®ç‚¹ã€‚è¿™æ˜¯ä¸€ç§æœ‰æ•ˆçš„æ­£åˆ™åŒ–æ–¹æ³•ï¼Œå¹¶å‡å°‘äº†è®­ç»ƒæ¨¡å‹æ‰€éœ€çš„å†…å­˜éœ€æ±‚ï¼Œè¿™æ˜¯æˆ‘ä»¬å°†æ•´ä¸ªæ¨¡å‹æ”¾å…¥GPUå†…å­˜æ‰€å¿…éœ€çš„ï¼ˆæ‰€æœ‰æ•°æ®å’Œä»£ç éƒ½å‘å¸ƒåœ¨é¡¹ç›®ç½‘ç«™ä¸Šï¼‰ã€‚

## 5 Experimental setup

**å®éªŒè£…ç½®**

We view the problem of predicting polypharmacy side effects as solving a multirelational link prediction task. Here, every drug pair is connected through zero, one or more relation types (i.e. side effect types) from a set of all relation types (i.e. all side effect types, see Section 2 and Fig. 1).

æˆ‘ä»¬å°†é¢„æµ‹å¤šè¯å‰¯ä½œç”¨çš„é—®é¢˜è§†ä¸ºè§£å†³å¤šå…³ç³»é“¾æ¥é¢„æµ‹ä»»åŠ¡ã€‚è¿™é‡Œï¼Œæ¯ä¸ªè¯ç‰©å¯¹é€šè¿‡ä¸€ç»„æ‰€æœ‰å…³ç³»ç±»å‹ï¼ˆå³æ‰€æœ‰å‰¯ä½œç”¨ç±»å‹ï¼Œå‚è§ç¬¬2èŠ‚å’Œå›¾1ï¼‰ä¸­çš„é›¶ã€ä¸€ä¸ªæˆ–å¤šä¸ªå…³ç³»ç±»å‹ï¼ˆå³å‰¯ä½œç”¨ç±»å‹ï¼‰è¿æ¥ã€‚

For each polypharmacy side effect type, we split drug pairs associated with that side effect into training, validation and test sets, ensuring that the validation and test sets each include 10% of drug pairs. For each side effect type, we use 80% of drug pairs to train a model, and 10% of drug pairs to select model parameters. The task is then to predict pairs of drugs that are associated with each side effect type. Note that we are extremely careful that there is information leakage between the folds and that the cross-validation is fair.

å¯¹äºæ¯ç§å¤šè¯å‰¯ä½œç”¨ç±»å‹ï¼Œæˆ‘ä»¬å°†ä¸è¯¥å‰¯ä½œç”¨ç›¸å…³çš„è¯ç‰©å¯¹åˆ†ä¸ºåŸ¹è®­ã€éªŒè¯å’Œæµ‹è¯•é›†ï¼Œç¡®ä¿éªŒè¯å’Œæµ‹è¯•é›†å‡åŒ…æ‹¬10%çš„è¯ç‰©å¯¹ã€‚å¯¹äºæ¯ç§å‰¯ä½œç”¨ç±»å‹ï¼Œæˆ‘ä»¬ä½¿ç”¨80%çš„è¯ç‰©å¯¹è®­ç»ƒæ¨¡å‹ï¼Œ10%çš„è¯ç‰©å¯¹é€‰æ‹©æ¨¡å‹å‚æ•°ã€‚æ¥ä¸‹æ¥çš„ä»»åŠ¡æ˜¯é¢„æµ‹ä¸æ¯ç§å‰¯ä½œç”¨ç±»å‹ç›¸å…³çš„è¯ç‰©å¯¹ã€‚è¯·æ³¨æ„ï¼Œæˆ‘ä»¬éå¸¸å°å¿ƒè¤¶çš±ä¹‹é—´å­˜åœ¨ä¿¡æ¯æ³„æ¼ï¼Œäº¤å‰éªŒè¯æ˜¯å…¬å¹³çš„ã€‚

We apply *Decagon*, which for every drug pair and for every side effect type calculates a probability that a given drug pair is associated with a given side effect. Additionally, we integrate side information, i.e. side effects of individual drugs (Section 2), into the model in the form of additional features **x**i for drug nodes *i*. To prevent any circularity and information leakage in the evaluation, we make sure that: (i) side effects we are predicting over are true polypharmacy side effects (i.e. a given polypharmacy side effect is only associated with a drug pair and not with any individual drug in the pair) and (ii) no side effect types that we are predicting over are included in the side features. For example, nausea is one polypharmacy side effect, and we therefore remove all instances of nausea as a side effect for individual drugs. We note that this is a conservative approach which allows us to reliably estimate prediction performance.

æˆ‘ä»¬ä½¿ç”¨*Decagon*ï¼Œå®ƒå¯¹æ¯ä¸€ç§è¯ç‰©å¯¹å’Œæ¯ä¸€ç§å‰¯ä½œç”¨ç±»å‹è®¡ç®—ä¸€ç§ç»™å®šè¯ç‰©å¯¹ä¸ç»™å®šå‰¯ä½œç”¨ç›¸å…³çš„æ¦‚ç‡ã€‚æ­¤å¤–ï¼Œæˆ‘ä»¬å°†å‰¯ä½œç”¨ä¿¡æ¯ï¼Œå³å•ä¸ªè¯ç‰©çš„å‰¯ä½œç”¨ï¼ˆç¬¬2èŠ‚ï¼‰ä»¥é™„åŠ ç‰¹å¾**x**içš„å½¢å¼é›†æˆåˆ°æ¨¡å‹ä¸­ï¼Œç”¨äºè¯ç‰©èŠ‚ç‚¹*i*ã€‚ä¸ºäº†é˜²æ­¢è¯„ä¼°ä¸­å‡ºç°ä»»ä½•å¾ªç¯å’Œä¿¡æ¯æ³„æ¼ï¼Œæˆ‘ä»¬ç¡®ä¿ï¼šï¼ˆiï¼‰æˆ‘ä»¬é¢„æµ‹çš„å‰¯ä½œç”¨æ˜¯çœŸæ­£çš„å¤šè¯å‰¯ä½œç”¨ï¼ˆå³ï¼Œç»™å®šçš„å¤šè¯å‰¯ä½œç”¨ä»…ä¸ä¸€å¯¹è¯ç‰©ç›¸å…³ï¼Œè€Œä¸è¯¥å¯¹è¯ç‰©ä¸­çš„ä»»ä½•å•ä¸ªè¯ç‰©æ— å…³ï¼‰å’Œï¼ˆiiï¼‰æˆ‘ä»¬é¢„æµ‹çš„å‰¯ä½œç”¨ç±»å‹æ²¡æœ‰åŒ…å«åœ¨å‰¯ä½œç”¨ç‰¹å¾ä¸­ã€‚ä¾‹å¦‚ï¼Œæ¶å¿ƒæ˜¯ä¸€ç§å¤šè¯å‰¯ä½œç”¨ï¼Œå› æ­¤æˆ‘ä»¬æ’é™¤äº†æ‰€æœ‰æ¶å¿ƒä½œä¸ºå•ä¸ªè¯ç‰©å‰¯ä½œç”¨çš„æƒ…å†µã€‚æˆ‘ä»¬æ³¨æ„åˆ°ï¼Œè¿™æ˜¯ä¸€ç§ä¿å®ˆçš„æ–¹æ³•ï¼Œå…è®¸æˆ‘ä»¬å¯é åœ°ä¼°è®¡é¢„æµ‹æ€§èƒ½ã€‚

We are not aware of any other approach developed for predicting side effects of drug pairs. We thus evaluate the performance of *Decagon* against the following multirelational link prediction approaches:

æˆ‘ä»¬è¿˜æ²¡æœ‰å‘ç°ä»»ä½•å…¶ä»–é¢„æµ‹è¯ç‰©å‰¯ä½œç”¨çš„æ–¹æ³•ã€‚å› æ­¤ï¼Œæˆ‘ä»¬æ ¹æ®ä»¥ä¸‹å¤šå…³ç³»é“¾è·¯é¢„æµ‹æ–¹æ³•è¯„ä¼°*Decagon*çš„æ€§èƒ½ï¼š

**RESCAL tensor decomposition** (Nickel *et al.*, 2011): This is a tensor factorization approach that takes a multirelational structure into account. Given **X**iâ , a drug-drug matrix encoding associations of drugs pairs with side effect *r*, matrix **X**i is decomposed as: **X**r=**A**T**r**A**T for r=1,2,â€¦,964â , where **T**r and **A** are model parameters. Given drugs *i* and *j*, their association with *r* is predicted as: **a**i**T**r**a**jâ .

**RESCALå¼ é‡åˆ†è§£**ï¼ˆNickel*ç­‰äººï¼Œ2011ï¼‰ï¼šè¿™æ˜¯ä¸€ç§è€ƒè™‘å¤šå…³ç³»ç»“æ„çš„å¼ é‡åˆ†è§£æ–¹æ³•ã€‚ç»™å®š**X**iâ , è¯ç‰©çŸ©é˜µç¼–ç å…·æœ‰å‰¯ä½œç”¨çš„è¯ç‰©å¯¹çš„å…³è”*r*ï¼ŒçŸ©é˜µ**X**iåˆ†è§£ä¸ºï¼š**X**r=**a**T**r**a**Tï¼Œr=1,2ï¼Œâ€¦ï¼Œ964â , å…¶ä¸­**T**rå’Œ**A**ä¸ºæ¨¡å‹å‚æ•°ã€‚ç»™å®šè¯ç‰©*i*å’Œ*j*ï¼Œå®ƒä»¬ä¸*r*çš„å…³è”é¢„æµ‹ä¸ºï¼š*a**i**T**r**a**jâ .

DEDICOM tensor decomposition(Papalexakis *et al.*, 2017): This is a related tensor factorization approach suitable for sparse data settings. A given drugâ€“drug matrix**X**i is decomposed as: **X**r=**A****U**r**T****U**r**A**Tâ . Given drugs *i* and *j*, their association with *r* is predicted as: **a**i**U**r**T****U**r**a**jâ .

DEDICOMå¼ é‡åˆ†è§£ï¼ˆPapalexakis*ç­‰äººï¼Œ2017ï¼‰ï¼šè¿™æ˜¯ä¸€ç§é€‚ç”¨äºç¨€ç–æ•°æ®è®¾ç½®çš„ç›¸å…³å¼ é‡åˆ†è§£æ–¹æ³•ã€‚ç»™å®šè¯ç‰©-è¯ç‰©çŸ©é˜µ**X**iåˆ†è§£ä¸ºï¼š**X**r=**A****U**r**T****U**r**A**Tâ . ç»™å®šè¯ç‰©*i*å’Œ*j*ï¼Œå®ƒä»¬ä¸*r*çš„å…³è”é¢„æµ‹ä¸ºï¼š*a**i**U**r**T****U**r**a**jâ .

**DeepWalk neural embeddings** (Perozzi *et al.*, 2014; Zong *et al.*, 2017): This approach learns *d*-dimensional neural features for nodes based on a biased random walk procedure exploring network neighborhoods of nodes. Drug pairs are represented by concatenating learned drug feature representations and used as input to a logistic regression classifier. For each link-type (i.e. side effect type), we train a separate logistic regression classifier.

**DeepWalkç¥ç»åµŒå…¥**ï¼ˆPerozzi*ç­‰äººã€‚*ï¼Œ2014ï¼›Zong*ç­‰äººã€‚*ï¼Œ2017ï¼‰ï¼šè¯¥æ–¹æ³•åŸºäºæ¢ç´¢èŠ‚ç‚¹ç½‘ç»œé‚»åŸŸçš„æœ‰åéšæœºæ¸¸èµ°ç¨‹åºå­¦ä¹ èŠ‚ç‚¹çš„*d*ç»´ç¥ç»ç‰¹å¾ã€‚è¯ç‰©å¯¹é€šè¿‡ä¸²è”å­¦ä¹ çš„è¯ç‰©ç‰¹å¾è¡¨ç¤ºæ¥è¡¨ç¤ºï¼Œå¹¶ç”¨ä½œé€»è¾‘å›å½’åˆ†ç±»å™¨çš„è¾“å…¥ã€‚å¯¹äºæ¯ä¸ªç¯èŠ‚ç±»å‹ï¼ˆå³å‰¯ä½œç”¨ç±»å‹ï¼‰ï¼Œæˆ‘ä»¬è®­ç»ƒä¸€ä¸ªå•ç‹¬çš„é€»è¾‘å›å½’åˆ†ç±»å™¨ã€‚

**Concatenated drug features**: This approach constructs a feature vector for each drug based on PCA representation of drugâ€“target protein interaction matrix and based on PCA representation of side effects of individual drugs. Drug pairs are represented by concatenating the corresponding drug feature vectors and used as input to a gradient boosting trees classifier that then predicts the exact side effect of a pair of drugs.

**ä¸²è”è¯ç‰©ç‰¹å¾**ï¼šè¯¥æ–¹æ³•åŸºäºè¯ç‰©-é¶è›‹ç™½ç›¸äº’ä½œç”¨çŸ©é˜µçš„PCAè¡¨ç¤ºå’Œå•ä¸ªè¯ç‰©å‰¯ä½œç”¨çš„PCAè¡¨ç¤ºï¼Œä¸ºæ¯ç§è¯ç‰©æ„å»ºä¸€ä¸ªç‰¹å¾å‘é‡ã€‚è¯ç‰©å¯¹é€šè¿‡è¿æ¥ç›¸åº”çš„è¯ç‰©ç‰¹å¾å‘é‡æ¥è¡¨ç¤ºï¼Œå¹¶ç”¨ä½œæ¢¯åº¦å¢å¼ºæ ‘åˆ†ç±»å™¨çš„è¾“å…¥ï¼Œç„¶åè¯¥åˆ†ç±»å™¨é¢„æµ‹è¯ç‰©å¯¹çš„ç¡®åˆ‡å‰¯ä½œç”¨ã€‚The parameter settings for every approach are determined using a validation set with a grid search over candidate parameter values (e.g. for gradient boosting trees, the number of trees used was varied from 10 to 100). In case an approach is not a multirelational link prediction method, we select parameters with best performance on the validation set individually for each side effect type. Specifically, *Decagon* uses a 2-layer neural architecture with d(1)=64â , and d(2)=32 hidden units in each layer, a dropout rate of 0.1, and a mini-batch size of 512 in all experiments.

Performance is calculated individually per side effect type using area under the receiver-operating characteristic (AUROC), area under the precision-recall curve (AUPRC) and average precision at 50 (AP@50). Higher values always indicate better performance.

ä½¿ç”¨æ¥æ”¶å™¨å·¥ä½œç‰¹æ€§ä¸‹çš„é¢ç§¯ï¼ˆAUROCï¼‰ã€ç²¾åº¦å¬å›æ›²çº¿ä¸‹çš„é¢ç§¯ï¼ˆAUPRCï¼‰å’Œ50%æ—¶çš„å¹³å‡ç²¾åº¦ï¼Œåˆ†åˆ«è®¡ç®—æ¯ç§å‰¯ä½œç”¨ç±»å‹çš„æ€§èƒ½(AP@50). å€¼è¶Šé«˜è¡¨ç¤ºæ€§èƒ½è¶Šå¥½ã€‚

## 6 Results

*Decagon* operates on multimodal graphs and in highly multirelational settings. This flexibility makes *Decagon* especially suitable for predicting side effects of pairs of drugs as we shall discuss below.

*Decagon*åœ¨å¤šæ¨¡æ€å›¾å’Œé«˜åº¦å¤šå…³ç³»è®¾ç½®ä¸­è¿è¡Œã€‚è¿™ç§çµæ´»æ€§ä½¿å¾—â€œDecagonâ€ç‰¹åˆ«é€‚åˆé¢„æµ‹æˆå¯¹è¯ç‰©çš„å‰¯ä½œç”¨ï¼Œæˆ‘ä»¬å°†åœ¨ä¸‹é¢è®¨è®ºã€‚

### 6.1 Prediction of polypharmacy side effects

**å¤šè¯å‰¯ä½œç”¨çš„é¢„æµ‹**

We start by comparing the performance of *Decagon* to alternative approaches. From results in Table 2, we see that considering the multimodal network representation and modeling a large number of different side effects allows *Decagon* to outperform other approaches by a large margin. Across 964 side effect types, *Decagon* outperforms alternative approaches by 19.7% (AUROC), 22.0% (AUPRC) and 36.3% (AP@50). *Decagon*â€™s improvement is especially pronounced relative to tensor factorization methods, where *Decagon* surpasses tensor-based methods by up to 68.7% (AP@50). This finding highlights a potential limitation of directly optimizing a tensor decomposition [i.e. vanilla RESCAL and DEDICOM (Nickel *et al.*, 2011; Papalexakis *et al.*, 2017)] without relying on a graph-structured convolutional encoder. We also compared *Decagon* with two other methods (Perozzi *et al.*, 2014; Zong *et al.*, 2017), which we adapted for a multirelational link prediction task. We observe that DeepWalk neural embeddings and Concatenated drug features achieve a gain of 9.0% (AUROC) and a 20.1% gain (AUPRC) over tensor-based methods. However, these approaches employ a two-stage pipeline, consisting of a drug feature extraction model and a link prediction model, both of which are trained separately. Furthermore, they cannot consider interdependence of different side effects that we showed to contain useful information (Section 3). These additional modeling insights, give *Decagon* a 22.0% gain over DeepWalk neural embeddings, and a 12.8% gain over Concatenated drug features in AP@50 scores.

æˆ‘ä»¬é¦–å…ˆæ¯”è¾ƒâ€œDecagonâ€ä¸å…¶ä»–æ–¹æ³•çš„æ€§èƒ½ã€‚ä»è¡¨2ä¸­çš„ç»“æœä¸­ï¼Œæˆ‘ä»¬å¯ä»¥çœ‹å‡ºï¼Œè€ƒè™‘åˆ°å¤šæ¨¡å¼ç½‘ç»œè¡¨ç¤ºå’Œå¯¹å¤§é‡ä¸åŒå‰¯ä½œç”¨çš„å»ºæ¨¡ï¼Œä½¿å¾—*Decagon*çš„è¡¨ç°å¤§å¤§ä¼˜äºå…¶ä»–æ–¹æ³•ã€‚åœ¨964ç§å‰¯ä½œç”¨ç±»å‹ä¸­ï¼Œ*Decagon*çš„è¡¨ç°ä¼˜äºå…¶ä»–æ–¹æ³•ï¼Œåˆ†åˆ«ä¸º19.7%ï¼ˆAUROCï¼‰ã€22.0%ï¼ˆAUPRCï¼‰å’Œ36.3%(AP@50). *Decagon*çš„æ”¹è¿›ç›¸å¯¹äºå¼ é‡å› å­åˆ†è§£æ–¹æ³•å°¤å…¶æ˜æ˜¾ï¼Œå…¶ä¸­Decagon*æ¯”åŸºäºå¼ é‡çš„æ–¹æ³•é«˜å‡º68.7%(AP@50). è¿™ä¸€å‘ç°çªå‡ºäº†ç›´æ¥ä¼˜åŒ–å¼ é‡åˆ†è§£çš„æ½œåœ¨å±€é™æ€§[å³vanilla RESCALå’ŒDEDICOMï¼ˆNickel*ç­‰ï¼Œ2011ï¼›Papalexakis*ç­‰ï¼Œ2017ï¼‰]ï¼Œè€Œä¸ä¾èµ–äºå›¾ç»“æ„å·ç§¯ç¼–ç å™¨ã€‚æˆ‘ä»¬è¿˜å°†*Decagon*ä¸å…¶ä»–ä¸¤ç§æ–¹æ³•ï¼ˆPerozzi*ç­‰ï¼Œ2014å¹´ï¼›Zong*ç­‰ï¼Œ2017å¹´ï¼‰è¿›è¡Œäº†æ¯”è¾ƒï¼Œè¿™ä¸¤ç§æ–¹æ³•é€‚ç”¨äºå¤šå…³ç³»é“¾è·¯é¢„æµ‹ä»»åŠ¡ã€‚æˆ‘ä»¬è§‚å¯Ÿåˆ°ï¼Œä¸åŸºäºå¼ é‡çš„æ–¹æ³•ç›¸æ¯”ï¼ŒDeepWalkç¥ç»åµŒå…¥å’Œä¸²è”è¯ç‰©ç‰¹å¾å®ç°äº†9.0%çš„å¢ç›Šï¼ˆAUROCï¼‰å’Œ20.1%çš„å¢ç›Šï¼ˆAUPRCï¼‰ã€‚ç„¶è€Œï¼Œè¿™äº›æ–¹æ³•é‡‡ç”¨ä¸¤çº§ç®¡é“ï¼ŒåŒ…æ‹¬è¯ç‰©ç‰¹å¾æå–æ¨¡å‹å’Œé“¾æ¥é¢„æµ‹æ¨¡å‹ï¼Œè¿™ä¸¤ä¸ªæ¨¡å‹éƒ½æ˜¯å•ç‹¬è®­ç»ƒçš„ã€‚æ­¤å¤–ï¼Œä»–ä»¬ä¸èƒ½è€ƒè™‘ä¸åŒçš„å‰¯ä½œç”¨ï¼Œæˆ‘ä»¬è¡¨ç°å‡ºåŒ…å«æœ‰ç”¨çš„ä¿¡æ¯ï¼ˆç¬¬3èŠ‚ï¼‰çš„ç›¸äº’ä¾èµ–æ€§ã€‚è¿™äº›é¢å¤–çš„å»ºæ¨¡æ´å¯Ÿï¼Œä½¿*Decagon*æ¯”DeepWalkç¥ç»åµŒå…¥å¢åŠ äº†22.0%ï¼Œæ¯”è”åˆè¯ç‰©ç‰¹å¾å¢åŠ äº†12.8%AP@50åˆ†æ•°ã€‚

**Table 2.**

Area under ROC curve (AUROC), area under precision-recall curve (AUPRC) and average precision at 50 (AP@50) for polypharmacy side effect prediction

ROCæ›²çº¿ä¸‹çš„é¢ç§¯ï¼ˆAUROCï¼‰ã€ç²¾ç¡®å›å¿†æ›²çº¿ä¸‹çš„é¢ç§¯ï¼ˆAUPRCï¼‰å’Œ50æ—¶çš„å¹³å‡ç²¾ç¡®åº¦(AP@50)ç”¨äºå¤šè¯å‰¯ä½œç”¨é¢„æµ‹

| Approach                     | AUROC | AUPRC | AP@50 |
| ---------------------------- | ----- | ----- | ----- |
| *Decagon*                    | 0.872 | 0.832 | 0.803 |
| RESCAL tensor factorization  | 0.693 | 0.613 | 0.476 |
| DEDICOM tensor factorization | 0.705 | 0.637 | 0.567 |
| DeepWalk neural embeddings   | 0.761 | 0.737 | 0.658 |
| Concatenated drug features   | 0.793 | 0.764 | 0.712 |

*Note*: Reported are average performance values for 964 side effect types.

These findings are aligned with results that predictions can often be significantly improved by end-to-end learning and specifically using graph auto-encoders (Hamilton *et al.*, 2017a, b; Kipf and Welling, 2016). In particular, tensor decomposition and neural embedding baseline approaches allow us to quantify what percentage of the performance improvement is due to the embeddings (i.e. *Decagon*â€™s encoder) and what percentage is due to the multitask learning (i.e. *Decagon*â€™s decoder).

è¿™äº›å‘ç°ä¸ä»¥ä¸‹ç»“æœä¸€è‡´ï¼šé€šè¿‡ç«¯åˆ°ç«¯å­¦ä¹ ï¼Œç‰¹åˆ«æ˜¯ä½¿ç”¨å›¾å½¢è‡ªåŠ¨ç¼–ç å™¨ï¼Œé¢„æµ‹é€šå¸¸å¯ä»¥æ˜¾è‘—æ”¹å–„ï¼ˆHamilton*ç­‰äººã€‚*ï¼Œ2017aï¼Œbï¼›Kipfå’ŒWellingï¼Œ2016ï¼‰ã€‚ç‰¹åˆ«æ˜¯ï¼Œå¼ é‡åˆ†è§£å’Œç¥ç»åµŒå…¥åŸºçº¿æ–¹æ³•å…è®¸æˆ‘ä»¬é‡åŒ–åµŒå…¥ï¼ˆå³*Decagon*çš„ç¼–ç å™¨ï¼‰å’Œå¤šä»»åŠ¡å­¦ä¹ ï¼ˆå³*Decagon*çš„è§£ç å™¨ï¼‰å¸¦æ¥çš„æ€§èƒ½æ”¹å–„ç™¾åˆ†æ¯”ã€‚

To better understand *Decagon*â€™s performance we stratify the aggregated statistics in Table 2 by side effect type. Manual examination of the results and a discussion with domain experts reveals a common property of best performing side effects in Table 3. We observe that *Decagon* models particularly well side effects with strong apparent molecular underpinnings. This observation is consistent with our expectation because *Decagon*â€™s multimodal graph (Fig. 1) contains predominantly pharmacogenomic information. We also observed that side effects with the worst performance tend to be common side effects and/or have non-molecular origins with potentially important environmental and behavioral components (Table 3). *Decagon*â€™s competitive performance on those side effects can be explained by effective sharing of model parameters across different types of side effects.

ä¸ºäº†æ›´å¥½åœ°ç†è§£*Decagon*çš„è¡¨ç°ï¼Œæˆ‘ä»¬å°†è¡¨2ä¸­çš„æ±‡æ€»ç»Ÿè®¡æ•°æ®æŒ‰å‰¯ä½œç”¨ç±»å‹è¿›è¡Œåˆ†å±‚ã€‚æ‰‹åŠ¨æ£€æŸ¥ç»“æœå¹¶ä¸é¢†åŸŸä¸“å®¶è®¨è®ºåï¼Œè¡¨3æ˜¾ç¤ºäº†æœ€ä½³å‰¯ä½œç”¨çš„å…±åŒç‰¹æ€§ã€‚æˆ‘ä»¬è§‚å¯Ÿåˆ°ï¼Œâ€œDecagonâ€æ¨¡å‹å…·æœ‰æ˜æ˜¾çš„åˆ†å­åŸºç¡€ï¼Œå‰¯ä½œç”¨ç‰¹åˆ«å¥½ã€‚è¿™ä¸€è§‚å¯Ÿç»“æœä¸æˆ‘ä»¬çš„é¢„æœŸä¸€è‡´ï¼Œå› ä¸º*Decagon*çš„å¤šæ¨¡æ€å›¾è°±ï¼ˆå›¾1ï¼‰ä¸»è¦åŒ…å«è¯ç‰©åŸºå› ç»„å­¦ä¿¡æ¯ã€‚æˆ‘ä»¬è¿˜è§‚å¯Ÿåˆ°ï¼Œè¡¨ç°æœ€å·®çš„å‰¯ä½œç”¨å¾€å¾€æ˜¯å¸¸è§çš„å‰¯ä½œç”¨å’Œ/æˆ–å…·æœ‰æ½œåœ¨é‡è¦ç¯å¢ƒå’Œè¡Œä¸ºæˆåˆ†çš„éåˆ†å­æ¥æºï¼ˆè¡¨3ï¼‰*Decagon*åœ¨è¿™äº›å‰¯ä½œç”¨æ–¹é¢çš„ç«äº‰è¡¨ç°å¯ä»¥é€šè¿‡æœ‰æ•ˆå…±äº«ä¸åŒç±»å‹å‰¯ä½œç”¨çš„æ¨¡å‹å‚æ•°æ¥è§£é‡Šã€‚

**Table 3.**

Side effects with the best and worst performance in *Decagon*

å‰¯ä½œç”¨åœ¨*Decagonä¸­è¡¨ç°æœ€ä½³å’Œæœ€å·®*

| **Best performing side effects** | **AUPRC** | **Worst performing side effects** | **AUPRC** |
| -------------------------------- | --------- | --------------------------------- | --------- |
| Mumps                            | 0.964     | Bleeding                          | 0.679     |
| Carbuncle                        | 0.949     | Increased body temp.              | 0.680     |
| Coccydynia                       | 0.943     | Emesis                            | 0.693     |
| Tympanic membrane perfor.        | 0.941     | Renal disorder                    | 0.694     |
| Dyshidrosis                      | 0.938     | Leucopenia                        | 0.695     |
| Spondylosis                      | 0.929     | Diarrhea                          | 0.705     |
| Schizoaffective disorder         | 0.919     | Icterus                           | 0.707     |
| Breast dysplasia                 | 0.918     | Nausea                            | 0.711     |
| Ganglion                         | 0.909     | Itch                              | 0.712     |
| Uterine polyp                    | 0.908     | Anaemia                           | 0.712     |
### 6.2 Investigation of *Decagon*â€™s novel predictions

decagoné¢„æµ‹ç ”ç©¶

Next, we perform a literature-based evaluation of new hits. Our goal is to evaluate the quality of novel *Decagon*â€™s predictions about relationships between side effects and drug pairs. To this aim, we ask *Decagon* to make a prediction for every drug pair and every side effect type in the dataset. We then use these predictions to construct a ranked list of (drug *i*, side effect type *r*, drug *j*) triples, where the triples are ranked by predicted probability scores pijr ([Equation 3](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E3)). We then exclude from the ranked list all the known associations between drug pairs and side effects, and afterwards investigate the 10 highest ranked predictions in the list. To prevent the risk of investigative bias, we do not allow any crosstalk between different stages of the analysis. We then search biomedical literature to see if we can find supporting evidence for these novel predictions.

æ¥ä¸‹æ¥ï¼Œæˆ‘ä»¬å¯¹æ–°çš„çƒ­é—¨ä½œå“è¿›è¡ŒåŸºäºæ–‡çŒ®çš„è¯„ä¼°ã€‚æˆ‘ä»¬çš„ç›®æ ‡æ˜¯è¯„ä¼°novel*Decagon*å…³äºå‰¯ä½œç”¨å’Œè¯ç‰©å¯¹ä¹‹é—´å…³ç³»çš„é¢„æµ‹çš„è´¨é‡ã€‚ä¸ºæ­¤ï¼Œæˆ‘ä»¬è¦æ±‚Decagonå¯¹æ•°æ®é›†ä¸­çš„æ¯ç§è¯ç‰©å¯¹å’Œæ¯ç§å‰¯ä½œç”¨ç±»å‹è¿›è¡Œé¢„æµ‹ã€‚ç„¶åï¼Œæˆ‘ä»¬ä½¿ç”¨è¿™äº›é¢„æµ‹æ„å»ºï¼ˆè¯ç‰©*i*ã€å‰¯ä½œç”¨ç±»å‹*r*ã€è¯ç‰©*j*ï¼‰ä¸‰å…ƒç»„çš„æ’åºåˆ—è¡¨ï¼Œå…¶ä¸­ä¸‰å…ƒç»„é€šè¿‡é¢„æµ‹æ¦‚ç‡å¾—åˆ†pijrè¿›è¡Œæ’åºï¼ˆ[ç­‰å¼3](https://academic.oup.com/bioinformatics/article/34/13/i457/5045770#E3)). ç„¶åï¼Œæˆ‘ä»¬å°†æ‰€æœ‰å·²çŸ¥çš„è¯ç‰©å¯¹ä¸å‰¯ä½œç”¨ä¹‹é—´çš„å…³è”ä»æ’ååˆ—è¡¨ä¸­æ’é™¤ï¼Œç„¶åè°ƒæŸ¥åˆ—è¡¨ä¸­æ’åæœ€é«˜çš„10ä¸ªé¢„æµ‹ã€‚ä¸ºäº†é˜²æ­¢è°ƒæŸ¥åå·®çš„é£é™©ï¼Œæˆ‘ä»¬ä¸å…è®¸åœ¨åˆ†æçš„ä¸åŒé˜¶æ®µä¹‹é—´å­˜åœ¨ä»»ä½•ä¸²æ‰°ã€‚ç„¶åæˆ‘ä»¬æœç´¢ç”Ÿç‰©åŒ»å­¦æ–‡çŒ®ï¼Œçœ‹çœ‹æ˜¯å¦èƒ½æ‰¾åˆ°æ”¯æŒè¿™äº›æ–°é¢„æµ‹çš„è¯æ®ã€‚

Table 4 shows *Decagonâ€™*s predictions and literature evidence supporting these predictions. We were able to find literature evidence for five out of 10 highest ranked predicted side effects. That is, our method both correctly identified the drug pair as well as the side effect type for these highest ranked predictions. This result is remarkable because the predictions were specific and the supporting evidence was very unlikely to be found by random selection of drug pair and side effect associations. We note that the cited literature explicitly investigates interactions between the predicted drug pair and the predicted side effect. For example, *Decagon* signified the use Atorvastatin and Amlodipine can lead to muscle inflammation (Table 4, 8th highest ranked prediction). In fact, recent reports (e.g. Banakh *et al.*, 2017) have found injuries in muscle tissue due to presumed drug interactions of Atorvastatin with Amlodipine. *Decagon* also flagged a potential association between Pyrimethamine, an anti-microbial that, if taken alone, is effective in the treatment of malaria, and Aliskiren, a renin inhibitor, whose clinical trial was halted after discovered kidney complications (Parving *et al.*, 2012), suggesting an increased risk of cancer (1st highest ranked prediction). The analysis here demonstrates the potential of *Decagonâ€™*s predictions to facilitate the translational science and the discovery of novel (non)-efficacious drug combinations.

è¡¨4æ˜¾ç¤ºäº†â€œDecagonâ€çš„é¢„æµ‹å’Œæ”¯æŒè¿™äº›é¢„æµ‹çš„æ–‡çŒ®è¯æ®ã€‚æˆ‘ä»¬èƒ½å¤Ÿæ‰¾åˆ°10ä¸ªæœ€é«˜çº§åˆ«çš„é¢„æµ‹å‰¯ä½œç”¨ä¸­çš„5ä¸ªçš„æ–‡çŒ®è¯æ®ã€‚ä¹Ÿå°±æ˜¯è¯´ï¼Œæˆ‘ä»¬çš„æ–¹æ³•æ—¢æ­£ç¡®è¯†åˆ«äº†è¯ç‰©å¯¹ï¼Œä¹Ÿæ­£ç¡®è¯†åˆ«äº†è¿™äº›æ’åæœ€é«˜çš„é¢„æµ‹çš„å‰¯ä½œç”¨ç±»å‹ã€‚è¿™ä¸€ç»“æœæ˜¯æ˜¾è‘—çš„ï¼Œå› ä¸ºé¢„æµ‹æ˜¯å…·ä½“çš„ï¼Œå¹¶ä¸”éšæœºé€‰æ‹©è¯ç‰©å¯¹å’Œå‰¯ä½œç”¨å…³è”ä¸å¤ªå¯èƒ½æ‰¾åˆ°æ”¯æŒè¯æ®ã€‚æˆ‘ä»¬æ³¨æ„åˆ°å¼•ç”¨çš„æ–‡çŒ®æ˜ç¡®è°ƒæŸ¥äº†é¢„æµ‹çš„è¯ç‰©å¯¹å’Œé¢„æµ‹çš„å‰¯ä½œç”¨ä¹‹é—´çš„ç›¸äº’ä½œç”¨ã€‚ä¾‹å¦‚ï¼Œ*Decagon*è¡¨ç¤ºä½¿ç”¨é˜¿æ‰˜ä¼ä»–æ±€å’Œæ°¨æ°¯åœ°å¹³ä¼šå¯¼è‡´è‚Œè‚‰ç‚ç—‡ï¼ˆè¡¨4ï¼Œæ’åç¬¬8çš„é¢„æµ‹ï¼‰ã€‚äº‹å®ä¸Šï¼Œæœ€è¿‘çš„æŠ¥å‘Šï¼ˆå¦‚Banakh*ç­‰äººï¼Œ2017å¹´ï¼‰å‘ç°ï¼Œç”±äºé˜¿æ‰˜ä¼ä»–æ±€ä¸æ°¨æ°¯åœ°å¹³çš„å‡å®šè¯ç‰©ç›¸äº’ä½œç”¨ï¼Œè‚Œè‚‰ç»„ç»‡å—åˆ°æŸä¼¤*Decagon*è¿˜æŒ‡å‡ºä¹™èƒºå˜§å•¶ï¼ˆä¸€ç§æŠ—èŒè¯ç‰©ï¼Œå¦‚æœå•ç‹¬æœç”¨ï¼Œå¯¹æ²»ç–—ç–Ÿç–¾æœ‰æ•ˆï¼‰ä¸è‚¾ç´ æŠ‘åˆ¶å‰‚é˜¿åˆ©å‰ä»‘ï¼ˆå…¶ä¸´åºŠè¯•éªŒåœ¨å‘ç°è‚¾è„å¹¶å‘ç—‡ååœæ­¢ï¼‰ä¹‹é—´å­˜åœ¨æ½œåœ¨å…³è”ï¼ˆParving*ç­‰äººï¼Œ2012å¹´ï¼‰ï¼Œè¿™è¡¨æ˜ç™Œç—‡é£é™©å¢åŠ ï¼ˆæ’åç¬¬ä¸€çš„é¢„æµ‹ï¼‰ã€‚è¿™é‡Œçš„åˆ†æè¯æ˜äº†â€œDecagonâ€é¢„æµ‹ä¿ƒè¿›è½¬åŒ–ç§‘å­¦å’Œå‘ç°æ–°å‹ï¼ˆéï¼‰æœ‰æ•ˆè¯ç‰©ç»„åˆçš„æ½œåŠ›ã€‚

**Table 4.**

New polypharmacy side effect predictions given by (drug *i*, side effect type *r*, drug *j*) triples that were assigned the highest probability scores by *Decagon*

ï¼ˆè¯ç‰©*i*ã€å‰¯ä½œç”¨ç±»å‹*r*ã€è¯ç‰©*j*ï¼‰ä¸‰å€ä½“ç»™å‡ºçš„æ–°çš„å¤šè¯å‰¯ä½œç”¨é¢„æµ‹ï¼Œè¿™äº›ä¸‰å€ä½“è¢«*Decagonåˆ†é…äº†æœ€é«˜æ¦‚ç‡åˆ†æ•°*

| k    | **Polypharmacy effect r** | **Drug i**    | **Drug j**    | Evidence                |
| ---- | ------------------------- | ------------- | ------------- | ----------------------- |
| 1    | Sarcoma                   | Pyrimethamine | Aliskiren     | Stage *et al.* (2015)   |
| 4    | Breast disorder           | Tolcapone     | Pyrimethamine | Bicker *et al.* (2017)  |
| 6    | Renal tubular acidosis    | Omeprazole    | Amoxicillin   | Russo *et al.* (2016)   |
| 8    | Muscle inflammation       | Atorvastatin  | Amlodipine    | Banakh *et al.* (2017)  |
| 9    | Breast inflammation       | Aliskiren     | Tioconazole   | Parving *et al.* (2012) |

*Note*: For each prediction, we include its rank *k* in the ranked list of all predictions and literature evidence supporting existence of the predicted association.

*æ³¨*ï¼šå¯¹äºæ¯ä¸ªé¢„æµ‹ï¼Œæˆ‘ä»¬åœ¨æ‰€æœ‰é¢„æµ‹å’Œæ”¯æŒé¢„æµ‹å…³è”å­˜åœ¨çš„æ–‡çŒ®è¯æ®çš„æ’ååˆ—è¡¨ä¸­åŒ…æ‹¬å…¶æ’å*k*ã€‚

### 6.3 Exploration of *Decagonâ€™*s side effect embeddings

å…³äºdecagonå‰¯ä½œç”¨åµŒå…¥çš„æ¢è®¨

Finally, we are interested in knowing whether *Decagon* meets the design goals presented in Section 3. In particular, we test if *Decagon* can capture the interdependence of different side effect types revealed by our exploratory data analysis (second observation in Section 3). To this aim, we take diagonal matrices **D**râ , which specifically model the importance of interactions for each side effect type *r* in *Decagonâ€™*s multirelational link prediction (Section 4.2). We extract the diagonal from each **D**r and use it as a vector representation for side effect *r*. We embed these vector representations into a 2D space using t-SNE (Maaten and Hinton, 2008) and then visualize in Figure 4.

æœ€åï¼Œæˆ‘ä»¬æƒ³çŸ¥é“*Decagon*æ˜¯å¦æ»¡è¶³ç¬¬3èŠ‚ä¸­æå‡ºçš„è®¾è®¡ç›®æ ‡ã€‚ç‰¹åˆ«æ˜¯ï¼Œæˆ‘ä»¬æµ‹è¯•*Decagon*æ˜¯å¦èƒ½å¤Ÿæ•è·æ¢ç´¢æ€§æ•°æ®åˆ†æï¼ˆç¬¬3èŠ‚ç¬¬äºŒä¸ªè§‚å¯Ÿç»“æœï¼‰æ‰€æ­ç¤ºçš„ä¸åŒå‰¯ä½œç”¨ç±»å‹çš„ç›¸äº’ä¾èµ–æ€§ã€‚ä¸ºæ­¤ï¼Œæˆ‘ä»¬é‡‡ç”¨å¯¹è§’çŸ©é˜µ**D**râ , å…·ä½“æ¨¡æ‹Ÿäº†*Decagon'*så¤šå…³ç³»é“¾è·¯é¢„æµ‹ä¸­æ¯ä¸ªå‰¯ä½œç”¨ç±»å‹*r*çš„äº¤äº’é‡è¦æ€§ï¼ˆç¬¬4.2èŠ‚ï¼‰ã€‚æˆ‘ä»¬ä»æ¯ä¸ª**D**rä¸­æå–å¯¹è§’çº¿ï¼Œå¹¶å°†å…¶ç”¨ä½œå‰¯ä½œç”¨*r*çš„å‘é‡è¡¨ç¤ºã€‚æˆ‘ä»¬ä½¿ç”¨t-SNEï¼ˆMaatenå’ŒHintonï¼Œ2008ï¼‰å°†è¿™äº›çŸ¢é‡è¡¨ç¤ºåµŒå…¥2Dç©ºé—´ï¼Œç„¶ååœ¨å›¾4ä¸­å¯è§†åŒ–ã€‚

![](index.assets/bty294f4.jpeg)

Visualization of side effects in Decagon. The side effects are mapped to the 2D space using the t-SNE package (Maaten and Hinton, 2008) with learned side effect representations [Dr,â€‰r=1,2,â€¦,964, see Equation (2)] as input. Selected side effects are uterine polyp, pancreatitis, viral meningitis and thyroid disease. For each selected side effect, we highlight three side effects that most often co-occur with the selected side effect in the drug combination dataset

Decagonå‰¯ä½œç”¨çš„å¯è§†åŒ–ã€‚ä½¿ç”¨t-SNEè½¯ä»¶åŒ…ï¼ˆMaatenå’ŒHintonï¼Œ2008ï¼‰å°†å‰¯ä½œç”¨æ˜ å°„åˆ°2Dç©ºé—´ï¼Œå¹¶å­¦ä¹ å‰¯ä½œç”¨è¡¨ç¤º[Drï¼Œâ€‰r=1,2ï¼Œâ€¦ï¼Œ964ï¼Œè§ç­‰å¼ï¼ˆ2ï¼‰]ä½œä¸ºè¾“å…¥ã€‚é€‰å®šçš„å‰¯ä½œç”¨æ˜¯å­å®«æ¯è‚‰ã€èƒ°è…ºç‚ã€ç—…æ¯’æ€§è„‘è†œç‚å’Œç”²çŠ¶è…ºç–¾ç—…ã€‚å¯¹äºæ¯ä¸ªé€‰æ‹©çš„å‰¯ä½œç”¨ï¼Œæˆ‘ä»¬å¼ºè°ƒäº†ä¸‰ä¸ªå‰¯ä½œç”¨ï¼Œå®ƒä»¬é€šå¸¸ä¸è¯ç‰©ç»„åˆæ•°æ®é›†ä¸­é€‰æ‹©çš„å‰¯ä½œç”¨åŒæ—¶å‘ç”Ÿ

Figure 4 reveals the existence of clustering structure in side effectsâ€™ representations. Examining the figure, we observe that side effects embedded close together in the 2D space tend to co-occur in drug combinations. This observation indicates that *Decagon*infers similar matrices **D**r1 and **D**r2 for side effects *r*1 and *r*2 that appear together in many drug combinations. For example, the top three side effects that often appear together with uterine polyp side effect are: uterine bleeding, breast dysplasia and postmenopausal bleeding. Indeed, *Decagon* infers similar diagonal factors **D**r for all three side effects, resulting in localized projections in the 2D space (Fig. 4).

å›¾4æ­ç¤ºäº†å‰¯ä½œç”¨è¡¨ç¤ºä¸­é›†ç¾¤ç»“æ„çš„å­˜åœ¨ã€‚é€šè¿‡æŸ¥çœ‹è¯¥å›¾ï¼Œæˆ‘ä»¬è§‚å¯Ÿåˆ°åœ¨2Dç©ºé—´ä¸­ç´§å¯†åµŒå…¥çš„å‰¯ä½œç”¨å¾€å¾€åœ¨è¯ç‰©ç»„åˆä¸­åŒæ—¶å‘ç”Ÿã€‚è¿™ä¸€è§‚å¯Ÿç»“æœè¡¨æ˜ï¼Œ*Decagon*æ¨æ–­å‡ºåœ¨è®¸å¤šè¯ç‰©ç»„åˆä¸­åŒæ—¶å‡ºç°çš„å‰¯ä½œç”¨*r*1å’Œ*r*2çš„ç±»ä¼¼çŸ©é˜µ**D**r1å’Œ**D**r2ã€‚ä¾‹å¦‚ï¼Œä¸å­å®«æ¯è‚‰å‰¯ä½œç”¨ä¸€èµ·å‡ºç°çš„å‰ä¸‰å¤§å‰¯ä½œç”¨æ˜¯ï¼šå­å®«å‡ºè¡€ã€ä¹³è…ºå‘è‚²ä¸è‰¯å’Œç»ç»åå‡ºè¡€ã€‚äº‹å®ä¸Šï¼Œ*Decagon*ä¸ºæ‰€æœ‰ä¸‰ç§å‰¯ä½œç”¨æ¨æ–­å‡ºç±»ä¼¼çš„å¯¹è§’çº¿å› å­**D**rï¼Œä»è€Œåœ¨2Dç©ºé—´ä¸­äº§ç”Ÿå±€éƒ¨æŠ•å½±ï¼ˆå›¾4ï¼‰ã€‚

To test if the appealing pattern in Figure 4 holds true across many side effect types we proceed as follows. We compute average Euclidean distance between each side effectâ€™s vector representation and vector representations of three most frequently co-occurring side effects. We find that co-occurring/related side effects have significantly more similar representations (i.e. diagonal factors **D**râ ) than expected by chance (*P*-valueâ€‰=â€‰1eâˆ’34â , 2-sample KS test). We thus conclude that *Decagon* is able to meet the design goals of polypharmacy side effect modeling. Furthermore, the analysis here indicates that *Decagonâ€™*s multirelational link prediction model (Section 4.2) can capture interdependence of side effects present in drug combination data.

ä¸ºäº†æµ‹è¯•å›¾4ä¸­çš„å¸å¼•äººçš„æ¨¡å¼æ˜¯å¦é€‚ç”¨äºè®¸å¤šå‰¯ä½œç”¨ç±»å‹ï¼Œæˆ‘ä»¬è¿›è¡Œå¦‚ä¸‹æ“ä½œã€‚æˆ‘ä»¬è®¡ç®—æ¯ä¸ªå‰¯ä½œç”¨çš„å‘é‡è¡¨ç¤ºå’Œä¸‰ä¸ªæœ€å¸¸åŒæ—¶å‘ç”Ÿçš„å‰¯ä½œç”¨çš„å‘é‡è¡¨ç¤ºä¹‹é—´çš„å¹³å‡æ¬§æ°è·ç¦»ã€‚æˆ‘ä»¬å‘ç°ï¼Œå…±ç°/ç›¸å…³å‰¯ä½œç”¨å…·æœ‰æ›´ä¸ºç›¸ä¼¼çš„è¡¨ç°ï¼ˆå³å¯¹è§’çº¿å› å­**D**rï¼‰â ) æ„å¤–è¶…å‡ºé¢„æœŸï¼ˆ*P*-å€¼â€‰=â€‰1eâˆ’34â , å› æ­¤ï¼Œæˆ‘ä»¬å¾—å‡ºç»“è®ºï¼Œ*Decagon*èƒ½å¤Ÿæ»¡è¶³å¤šè¯å‰¯ä½œç”¨å»ºæ¨¡çš„è®¾è®¡ç›®æ ‡ã€‚æ­¤å¤–ï¼Œè¿™é‡Œçš„åˆ†æè¡¨æ˜ï¼Œ*Decagon'sçš„å¤šå…³ç³»é“¾æ¥é¢„æµ‹æ¨¡å‹ï¼ˆç¬¬4.2èŠ‚ï¼‰èƒ½å¤Ÿæ•è·è¯ç‰©ç»„åˆæ•°æ®ä¸­å‰¯ä½œç”¨çš„ç›¸äº’ä¾èµ–æ€§ã€‚

## 7 Related work

We review related research on computational prediction of drug combinations, and on neural networks for graph-structured data.

æˆ‘ä»¬å›é¡¾äº†è¯ç‰©ç»„åˆçš„è®¡ç®—é¢„æµ‹å’Œå›¾å½¢ç»“æ„æ•°æ®çš„ç¥ç»ç½‘ç»œçš„ç›¸å…³ç ”ç©¶ã€‚

### 7.1 Drug combination modeling

**è¯ç‰©ç»„åˆæ¨¡å‹**

Methods in computational pharmacology aim to find associations between drugs and molecular targets, predict potential adverse drug reactions and find new uses of existing drugs (Campillos *et al.*, 2008; Hodos *et al.*, 2016; Li *et al.*, 2016). In contrast to individual drugs and single drug therapy (i.e. monotherapy) predominantly considered by these methods, we consider drug combinations (i.e. polypharmacy). This is important as polypharmacy is a useful strategy for combating complex diseases (Han *et al.*, 2017; Jia *et al.*, 2009) with important implications for health care system (Ernst and Grizzle, 2001).

è®¡ç®—è¯ç†å­¦ä¸­çš„æ–¹æ³•æ—¨åœ¨å‘ç°è¯ç‰©ä¸åˆ†å­é¶ç‚¹ä¹‹é—´çš„å…³è”ï¼Œé¢„æµ‹æ½œåœ¨çš„è¯ç‰©ä¸è‰¯ååº”ï¼Œå¹¶å‘ç°ç°æœ‰è¯ç‰©çš„æ–°ç”¨é€”ï¼ˆCampillos*ç­‰ï¼Œ2008ï¼›Hodos*ç­‰ï¼Œ2016ï¼›Li*ç­‰ï¼Œ2016ï¼‰ã€‚ä¸ä¸ªåˆ«è¯ç‰©å’Œå•è¯æ²»ç–—ï¼ˆå³å•è¯ï¼‰ç›¸æ¯”ï¼Œè¿™äº›æ–¹æ³•ä¸»è¦è€ƒè™‘ï¼Œæˆ‘ä»¬è€ƒè™‘è¯ç‰©ç»„åˆï¼ˆå³å¤šè¯ï¼‰ã€‚è¿™ä¸€ç‚¹å¾ˆé‡è¦ï¼Œå› ä¸ºå¤šè¯ç–—æ³•æ˜¯å¯¹æŠ—å¤æ‚ç–¾ç—…çš„æœ‰æ•ˆç­–ç•¥ï¼ˆHan*ç­‰äººï¼Œ2017å¹´ï¼›Jia*ç­‰äººï¼Œ2009å¹´ï¼‰ï¼Œå¯¹åŒ»ç–—ä½“ç³»å…·æœ‰é‡è¦å½±å“ï¼ˆErnstå’ŒGrizzleï¼Œ2001å¹´ï¼‰ã€‚

Traditionally, effective drug combinations have been identified by experimentally screening all possible combinations of a pre-defined set of drugs (Chen *et al.*, 2016b). Given the large number of drugs, experimental screens of pairwise combinations of drugs pose a formidable challenge in terms of cost and time. For example, given *n* drugs, there are n(nâˆ’1)/2 pairwise drug combinations and many more higher-order combinations. To address the combinatorial explosion of candidate drug combinations, computational methods were developed to identify drug pairs that potentially interact, i.e. drug pairs that produce an exaggerated response over and beyond the additive response expected under no interaction (Ryall and Tan, 2015). Previous research in this realm focused on defining drugâ€“drug interactions through the concepts of synergy and antagonism (Lewis *et al.*, 2015; Loewe, 1953), quantitatively measuring dose-effect curves (Bansal *et al.*, 2014; Takeda *et al.*, 2017) and determining whether or not a given drug pair interacts according to an experiment measuring cell viability (Chen *et al.*, 2016a, b; Huang *et al.*, 2014a, b; Shi *et al.*, 2017; Sun *et al.*, 2015; Zitnik and Zupan, 2016). All of these approaches predict drugâ€“drug interactions as scalar values representing the overall probability/strength of an interaction for a given drug pair. In sharp contrast, our study here goes a step further and identifies how exactly, if at all, a given drug pair manifests clinically within a patient population. In particular, we model clinical manifestations that cannot be attributed to either drug alone and that arise due to drug interaction (i.e. polypharmacy side effects). Whereas previous research focused on generating pointwise interaction estimates representing cell viability or a closely related outcome in an experimental drug screen, we predict, for the first time, which, if any, polypharmacy side effects can occur when multiples drugs are taken together by a patient, yielding a more direct path for clinical translation.

ä¼ ç»Ÿä¸Šï¼Œé€šè¿‡å®éªŒç­›é€‰ä¸€ç»„é¢„å®šä¹‰è¯ç‰©çš„æ‰€æœ‰å¯èƒ½ç»„åˆæ¥ç¡®å®šæœ‰æ•ˆè¯ç‰©ç»„åˆï¼ˆChen*ç­‰äººã€‚*ï¼Œ2016bï¼‰ã€‚é‰´äºè¯ç‰©æ•°é‡å·¨å¤§ï¼Œè¯ç‰©æˆå¯¹ç»„åˆçš„å®éªŒç­›é€‰åœ¨æˆæœ¬å’Œæ—¶é—´ä¸Šéƒ½æ˜¯ä¸€ä¸ªå·¨å¤§çš„æŒ‘æˆ˜ã€‚ä¾‹å¦‚ï¼Œç»™å®š*n*ç§è¯ç‰©ï¼Œå°±æœ‰nï¼ˆnâˆ’1ï¼‰ /2æˆå¯¹è¯ç‰©ç»„åˆå’Œæ›´å¤šé«˜é˜¶ç»„åˆã€‚ä¸ºäº†è§£å†³å€™é€‰è¯ç‰©ç»„åˆçš„ç»„åˆçˆ†ç‚¸é—®é¢˜ï¼Œå¼€å‘äº†è®¡ç®—æ–¹æ³•ï¼Œä»¥ç¡®å®šå¯èƒ½ç›¸äº’ä½œç”¨çš„è¯ç‰©å¯¹ï¼Œå³åœ¨æ— ç›¸äº’ä½œç”¨çš„æƒ…å†µä¸‹äº§ç”Ÿè¶…å‡ºé¢„æœŸåŠ æ€§ååº”çš„å¤¸å¼ ååº”çš„è¯ç‰©å¯¹ï¼ˆRyallå’ŒTanï¼Œ2015ï¼‰ã€‚è¯¥é¢†åŸŸå…ˆå‰çš„ç ”ç©¶ä¾§é‡äºé€šè¿‡ååŒå’Œæ‹®æŠ—æ¦‚å¿µå®šä¹‰è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ï¼ˆLewis*ç­‰ï¼Œ2015ï¼›Loeweï¼Œ1953ï¼‰ï¼Œå®šé‡æµ‹é‡å‰‚é‡-æ•ˆåº”æ›²çº¿ï¼ˆBansal*ç­‰ï¼Œ2014ï¼›Takeda*ç­‰ï¼Œ2017ï¼‰å¹¶æ ¹æ®æµ‹é‡ç»†èƒæ´»åŠ›çš„å®éªŒç¡®å®šç»™å®šè¯ç‰©å¯¹æ˜¯å¦ç›¸äº’ä½œç”¨ï¼ˆChen*ç­‰äººã€‚*ï¼Œ2016aï¼Œbï¼›Huang*ç­‰äººã€‚*ï¼Œ2014aï¼Œbï¼›Shi*ç­‰äººã€‚*ï¼Œ2017ï¼›Sun*ç­‰äººã€‚*ï¼Œ2015ï¼›Zitnikå’ŒZupanï¼Œ2016ï¼‰ã€‚æ‰€æœ‰è¿™äº›æ–¹æ³•éƒ½å°†è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨é¢„æµ‹ä¸ºä»£è¡¨ç»™å®šè¯ç‰©å¯¹ç›¸äº’ä½œç”¨çš„æ€»ä½“æ¦‚ç‡/å¼ºåº¦çš„æ ‡é‡å€¼ã€‚ä¸æ­¤å½¢æˆé²œæ˜å¯¹æ¯”çš„æ˜¯ï¼Œæˆ‘ä»¬çš„ç ”ç©¶æ›´è¿›ä¸€æ­¥ï¼Œç¡®å®šäº†ç»™å®šè¯ç‰©å¯¹åœ¨æ‚£è€…ç¾¤ä½“ä¸­çš„ä¸´åºŠè¡¨ç°ã€‚ç‰¹åˆ«æ˜¯ï¼Œæˆ‘ä»¬æ¨¡æ‹Ÿäº†ä¸èƒ½å•ç‹¬å½’å› äºä¸¤ç§è¯ç‰©çš„ä¸´åºŠè¡¨ç°ï¼Œä»¥åŠç”±äºè¯ç‰©ç›¸äº’ä½œç”¨ï¼ˆå³å¤šè¯å‰¯ä½œç”¨ï¼‰è€Œäº§ç”Ÿçš„ä¸´åºŠè¡¨ç°ã€‚é‰´äºä¹‹å‰çš„ç ”ç©¶ä¾§é‡äºåœ¨å®éªŒè¯ç‰©ç­›é€‰ä¸­äº§ç”Ÿä»£è¡¨ç»†èƒæ´»åŠ›æˆ–å¯†åˆ‡ç›¸å…³ç»“æœçš„é€ç‚¹ç›¸äº’ä½œç”¨ä¼°è®¡ï¼Œæˆ‘ä»¬é¦–æ¬¡é¢„æµ‹ï¼Œå½“æ‚£è€…æœç”¨å¤šç§è¯ç‰©æ—¶ï¼Œå¯èƒ½ä¼šå‡ºç°å“ªç§ï¼ˆå¦‚æœæœ‰çš„è¯ï¼‰å¤šè¯å‰¯ä½œç”¨ï¼Œä¸ºä¸´åºŠç¿»è¯‘æä¾›æ›´ç›´æ¥çš„é€”å¾„ã€‚

Although present drugâ€“drug interaction prediction approaches cannot be directly used for the problem studied here, we briefly overview methodology used by these approaches. Drugâ€“drug interaction prediction approaches can be categorized into classification-based and similarity-based methods. Classification-based methods consider drugâ€“drug interaction prediction as a binary classification problem (Chen *et al.*, 2016b; Cheng and Zhao, 2014; Huang *et al.*, 2014a; Shi *et al.*, 2017; Zitnik and Zupan, 2016). These methods use known interacting drug pairs as positive examples and other drug pairs as negative examples, and train classification models, such as naive Bayes, logistic regression and support vector machine. In contrast, similarity-based methods assume that similar drugs may have similar interaction patterns (Gottlieb *et al.*, 2012; Huang *et al.*, 2014b; Li *et al.*, 2016, 2017; Sun *et al.*, 2015; Vilar *et al.*, 2012; Zitnik and Zupan, 2015). These methods use different kinds of drugâ€“drug similarity measures defined on drug chemical substructures, interaction profile fingerprints, drug side effects, off-side effects and connectivity of molecular targets. The methods aggregate similarity measures through clustering or label propagation in order to identify potential drugâ€“drug interactions (Ferdousi *et al.*, 2017; Zhang *et al.*, 2015, 2017). However, all these methods generate drugâ€“drug interaction scores and do not predict the exact polypharmacy side effect, which is the goal of our study here.

è™½ç„¶ç›®å‰çš„è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨é¢„æµ‹æ–¹æ³•ä¸èƒ½ç›´æ¥ç”¨äºè¿™é‡Œç ”ç©¶çš„é—®é¢˜ï¼Œä½†æˆ‘ä»¬ç®€è¦æ¦‚è¿°äº†è¿™äº›æ–¹æ³•ä½¿ç”¨çš„æ–¹æ³•ã€‚è¯ç‰©ç›¸äº’ä½œç”¨é¢„æµ‹æ–¹æ³•å¯åˆ†ä¸ºåŸºäºåˆ†ç±»çš„æ–¹æ³•å’ŒåŸºäºç›¸ä¼¼æ€§çš„æ–¹æ³•ã€‚åŸºäºåˆ†ç±»çš„æ–¹æ³•è®¤ä¸ºè¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨é¢„æµ‹æ˜¯äºŒå…ƒåˆ†ç±»é—®é¢˜ï¼ˆé™ˆ*ç­‰ï¼Œ2016Bï¼›ç¨‹å’Œèµµï¼Œ2014ï¼›é»„*ç­‰äººï¼Œ*ï¼Œ2014Aï¼›S*ç­‰ï¼Œ2017ï¼›ZITNIKå’ŒZUPANNï¼Œ2016ï¼‰ã€‚è¿™äº›æ–¹æ³•ä½¿ç”¨å·²çŸ¥çš„ç›¸äº’ä½œç”¨è¯ç‰©å¯¹ä½œä¸ºæ­£ä¾‹ï¼Œå…¶ä»–è¯ç‰©å¯¹ä½œä¸ºè´Ÿä¾‹ï¼Œå¹¶è®­ç»ƒåˆ†ç±»æ¨¡å‹ï¼Œå¦‚æœ´ç´ è´å¶æ–¯ã€é€»è¾‘å›å½’å’Œæ”¯æŒå‘é‡æœºã€‚ç›¸åï¼ŒåŸºäºç›¸ä¼¼æ€§çš„æ–¹æ³•å‡è®¾ç›¸ä¼¼çš„è¯ç‰©å¯èƒ½å…·æœ‰ç›¸ä¼¼çš„ç›¸äº’ä½œç”¨æ¨¡å¼ï¼ˆGottlieb*ç­‰ï¼Œ2012ï¼›Huang*ç­‰ï¼Œ2014bï¼›Li*ç­‰ï¼Œ2016ï¼Œ2017ï¼›Sun*ç­‰ï¼Œ2015ï¼›Vilar*ç­‰ï¼Œ2012ï¼›Zitnikå’ŒZupanï¼Œ2015ï¼‰ã€‚è¿™äº›æ–¹æ³•ä½¿ç”¨ä¸åŒç§ç±»çš„è¯ç‰©-è¯ç‰©ç›¸ä¼¼æ€§åº¦é‡ï¼Œè¿™äº›åº¦é‡å®šä¹‰åœ¨è¯ç‰©åŒ–å­¦äºšç»“æ„ã€ç›¸äº’ä½œç”¨å›¾è°±æŒ‡çº¹ã€è¯ç‰©å‰¯ä½œç”¨ã€å‰¯ä½œç”¨å’Œåˆ†å­é¶ç‚¹çš„è¿æ¥æ€§ä¸Šã€‚è¿™äº›æ–¹æ³•é€šè¿‡èšç±»æˆ–æ ‡ç­¾ä¼ æ’­èšåˆç›¸ä¼¼æ€§åº¦é‡ï¼Œä»¥è¯†åˆ«æ½œåœ¨çš„è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨ï¼ˆFerdousi*ç­‰ï¼Œ2017ï¼›Zhang*ç­‰ï¼Œ2015ï¼Œ2017ï¼‰ã€‚ç„¶è€Œï¼Œæ‰€æœ‰è¿™äº›æ–¹æ³•éƒ½ä¼šäº§ç”Ÿè¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨è¯„åˆ†ï¼Œå¹¶ä¸”ä¸èƒ½é¢„æµ‹ç¡®åˆ‡çš„å¤šè¯å‰¯ä½œç”¨ï¼Œè¿™æ˜¯æˆ‘ä»¬åœ¨è¿™é‡Œç ”ç©¶çš„ç›®æ ‡ã€‚

### 7.2 Neural networks on graphs

**å›¾ä¸Šçš„ç¥ç»ç½‘ç»œ**

Our model extends existing work in the field of neural networks on graphs (Defferrard *et al.*, 2016; Gilmer *et al.*, 2017; Hamilton *et al.*, 2017a, b; Kipf and Welling, 2016; Schlichtkrull *et al.*, 2017). Neural networks on graphs enable learning over graph structures by generalizing the notion of convolution operation typically applied to image datasets to operations that can operate on arbitrary graphs. These neural networks can also be seen as an embedding methodology that distills high-dimensional information about each nodeâ€™s neighborhood into a dense vector embedding without requiring manual feature engineering. In particular, graph convolutional networks (Defferrard *et al.*, 2016; Hamilton *et al.*, 2017a; Kipf and Welling, 2016) and message passing neural networks (Gilmer *et al.*, 2017) are related lines of research that allow for layer-wise learning of node embeddings in graphs.

æˆ‘ä»¬çš„æ¨¡å‹æ‰©å±•äº†å›¾å½¢ç¥ç»ç½‘ç»œé¢†åŸŸçš„ç°æœ‰å·¥ä½œï¼ˆDefferard*ç­‰ï¼Œ2016ï¼›Gilmer*ç­‰ï¼Œ2017ï¼›Hamilton*ç­‰ï¼Œ2017aï¼Œbï¼›Kipfå’ŒWellingï¼Œ2016ï¼›Schlichtkrull*ç­‰ï¼Œ2017ï¼‰ã€‚å›¾ä¸Šçš„ç¥ç»ç½‘ç»œé€šè¿‡å°†é€šå¸¸åº”ç”¨äºå›¾åƒæ•°æ®é›†çš„å·ç§¯è¿ç®—çš„æ¦‚å¿µæ¨å¹¿åˆ°å¯æ“ä½œäºä»»æ„å›¾çš„è¿ç®—ï¼Œå®ç°äº†å¯¹å›¾ç»“æ„çš„å­¦ä¹ ã€‚è¿™äº›ç¥ç»ç½‘ç»œä¹Ÿå¯ä»¥çœ‹ä½œæ˜¯ä¸€ç§åµŒå…¥æ–¹æ³•ï¼Œå®ƒå°†æ¯ä¸ªèŠ‚ç‚¹é‚»åŸŸçš„é«˜ç»´ä¿¡æ¯æå–åˆ°å¯†é›†å‘é‡åµŒå…¥ä¸­ï¼Œè€Œæ— éœ€æ‰‹åŠ¨ç‰¹å¾å·¥ç¨‹ã€‚ç‰¹åˆ«æ˜¯ï¼Œå›¾å·ç§¯ç½‘ç»œï¼ˆDefferard*ç­‰äººã€‚*ï¼Œ2016ï¼›Hamilton*ç­‰äººã€‚*ï¼Œ2017aï¼›Kipfå’ŒWellingï¼Œ2016ï¼‰å’Œæ¶ˆæ¯ä¼ é€’ç¥ç»ç½‘ç»œï¼ˆGilmer*ç­‰äººã€‚*ï¼Œ2017ï¼‰æ˜¯ç›¸å…³çš„ç ”ç©¶çº¿ï¼Œå…è®¸å¯¹å›¾ä¸­çš„èŠ‚ç‚¹åµŒå…¥è¿›è¡Œåˆ†å±‚å­¦ä¹ ã€‚

Although graph convolutional networks achieve state-of-the-art performance on important prediction problems in social networks and knowledge graphs, they have not yet been used for problems in computational biology. Our model extends graph convolutional networks by incorporating support for multiple edge types, each type representing a different side effect, and by providing a form of efficient weight sharing for multimodal graphs with a large number of edge types.

å°½ç®¡å›¾å·ç§¯ç½‘ç»œåœ¨ç¤¾ä¼šç½‘ç»œå’ŒçŸ¥è¯†å›¾ä¸­çš„é‡è¦é¢„æµ‹é—®é¢˜ä¸Šå–å¾—äº†æœ€å…ˆè¿›çš„æ€§èƒ½ï¼Œä½†å®ƒä»¬å°šæœªç”¨äºè®¡ç®—ç”Ÿç‰©å­¦ä¸­çš„é—®é¢˜ã€‚æˆ‘ä»¬çš„æ¨¡å‹æ‰©å±•äº†å›¾å·ç§¯ç½‘ç»œï¼ŒåŒ…æ‹¬æ”¯æŒå¤šç§è¾¹ç±»å‹ï¼Œæ¯ç§ç±»å‹ä»£è¡¨ä¸åŒçš„å‰¯ä½œç”¨ï¼Œå¹¶ä¸ºå…·æœ‰å¤§é‡è¾¹ç±»å‹çš„å¤šæ¨¡å›¾æä¾›äº†ä¸€ç§æœ‰æ•ˆçš„æƒé‡å…±äº«å½¢å¼ã€‚

## 8 Conclusion

We presented *Decagon*, an approach for predicting side effects of drug pairs. *Decagon* is a general graph convolutional neural network designed to operate on a large multimodal graph where nodes can be connected through a large number of different relation types. We use *Decagon* to, for the first time, infer a prediction model that can identify side effects of pairs of drugs. *Decagon* predicts an association between a side effect and a co-prescribed drug pair (i.e. a drug combination) to identify side effects that cannot be attributed to either drug alone. The graph convolutional model achieves excellent accuracy on the polypharmacy side effect prediction task, allows us to consider nearly a thousand different side effect types integrating molecular and patient population data, and provides insights into clinical manifestation of drugâ€“drug interactions.

æˆ‘ä»¬æå‡ºäº†ä¸€ç§é¢„æµ‹è¯ç‰©å¯¹å‰¯ä½œç”¨çš„æ–¹æ³•*Decagon**Decagon*æ˜¯ä¸€ç§é€šç”¨å›¾å·ç§¯ç¥ç»ç½‘ç»œï¼Œè®¾è®¡ç”¨äºå¤„ç†å¤§å‹å¤šæ¨¡å›¾ï¼Œå…¶ä¸­èŠ‚ç‚¹å¯ä»¥é€šè¿‡å¤§é‡ä¸åŒçš„å…³ç³»ç±»å‹è¿æ¥ã€‚æˆ‘ä»¬é¦–æ¬¡ä½¿ç”¨â€œDecagonâ€æ¨æ–­å‡ºä¸€ä¸ªé¢„æµ‹æ¨¡å‹ï¼Œè¯¥æ¨¡å‹å¯ä»¥è¯†åˆ«æˆå¯¹è¯ç‰©çš„å‰¯ä½œç”¨*Decagon*é¢„æµ‹å‰¯ä½œç”¨ä¸è”åˆå¤„æ–¹è¯å¯¹ï¼ˆå³è¯ç‰©ç»„åˆï¼‰ä¹‹é—´çš„å…³è”ï¼Œä»¥ç¡®å®šä¸èƒ½å•ç‹¬å½’å› äºä»»ä½•ä¸€ç§è¯ç‰©çš„å‰¯ä½œç”¨ã€‚å›¾å·ç§¯æ¨¡å‹åœ¨å¤šè¯å‰¯ä½œç”¨é¢„æµ‹ä»»åŠ¡ä¸Šå–å¾—äº†å¾ˆé«˜çš„å‡†ç¡®æ€§ï¼Œå…è®¸æˆ‘ä»¬è€ƒè™‘æ•´åˆåˆ†å­å’Œæ‚£è€…ç¾¤ä½“æ•°æ®çš„è¿‘ä¸€åƒç§ä¸åŒçš„å‰¯ä½œç”¨ç±»å‹ï¼Œå¹¶æä¾›è¯ç‰©-è¯ç‰©ç›¸äº’ä½œç”¨çš„ä¸´åºŠè¡¨ç°çš„è§è§£ã€‚

There are several directions for future study. Our approach integrates molecular proteinâ€“protein and drugâ€“target networks together with population-level patientsâ€™ side effect data. Other sources of biomedical information, such as dosed concentration levels of drugs, might be relevant for modeling side effects of drug pairs, and we hope to investigate the utility of integrating them into the model. As *Decagonâ€™*s graph convolutional model is a general approach for multirelational link prediction in any multimodal network, it would be interesting to apply it to other domains and problems, for example, finding associations between patient outcomes and comorbid diseases, or for identifying dependencies between mutant phenotypes and geneâ€“gene interactions.

æœªæ¥çš„ç ”ç©¶æœ‰å‡ ä¸ªæ–¹å‘ã€‚æˆ‘ä»¬çš„æ–¹æ³•å°†åˆ†å­è›‹ç™½è´¨-è›‹ç™½è´¨å’Œè¯ç‰©-é¶ç‚¹ç½‘ç»œä¸äººç¾¤æ°´å¹³æ‚£è€…çš„å‰¯ä½œç”¨æ•°æ®ç»“åˆèµ·æ¥ã€‚å…¶ä»–ç”Ÿç‰©åŒ»å­¦ä¿¡æ¯æ¥æºï¼Œå¦‚è¯ç‰©çš„å‰‚é‡æµ“åº¦æ°´å¹³ï¼Œå¯èƒ½ä¸è¯ç‰©å¯¹çš„å‰¯ä½œç”¨å»ºæ¨¡æœ‰å…³ï¼Œæˆ‘ä»¬å¸Œæœ›ç ”ç©¶å°†å®ƒä»¬æ•´åˆåˆ°æ¨¡å‹ä¸­çš„æ•ˆç”¨ã€‚ç”±äº*Decagon'*så›¾å·ç§¯æ¨¡å‹æ˜¯ä»»ä½•å¤šæ¨¡ç½‘ç»œä¸­å¤šå…³ç³»é“¾è·¯é¢„æµ‹çš„é€šç”¨æ–¹æ³•ï¼Œå› æ­¤å°†å…¶åº”ç”¨äºå…¶ä»–é¢†åŸŸå’Œé—®é¢˜å°†æ˜¯æœ‰è¶£çš„ï¼Œä¾‹å¦‚ï¼Œå‘ç°æ‚£è€…ç»“æœä¸å…±ç—…ä¹‹é—´çš„å…³è”ï¼Œæˆ–è€…ç”¨äºè¯†åˆ«çªå˜è¡¨å‹å’ŒåŸºå› -åŸºå› ç›¸äº’ä½œç”¨ä¹‹é—´çš„ä¾èµ–æ€§ã€‚
